
% Summary paragraph
Recent advances in ARG inference have generated significant interest in leveraging ARGs for genetic analyses.
%
In this paper, we examined the relationship between different definitions of genetic relatedness
in the common context of additive traits on an ARG,
especially the emergent notion of ``branch'' relatedness.
%
We also demonstrated how branch relatedness compares with pedigree relatedness in simulations through a pedigree of French-Canadians.
%
We then described an algorithm for branch GRM-vector product, to bypass the fundamental problem of
quadratic complexity of computing and storing GRMs.
%
This algorithm allowed us to use randomized linear algebra methods
to do branch PCA using an ARG on a million samples in 30 seconds
and less than 4GB of RAM.
%
We close the discussion with an outlook for the use of these algorithms in population and quantitative genomic analyses.

% Paragraph on key points behind branch GRM
% How we relate to \cite{fan2022genealogical} and \cite{zhang2023biobank}
% and past work with pedigree, phylogenetic trees, and genotypes



\todo[inline]{Cite genetic vs genealogical/pedigree ancestors papers and why branch GRM can be more variable than pedigree GRM even when we do it across chromosomes - see results}

% TODO: discuss our observed variance in branch GRM and Fan et al. vGRM

A trait-centric perspective on relatedness is suggestive of a number of possible extensions. Although our current definition of relatedness assumes equal prior effects across all loci, one could consider alternatives whereby we incorporate prior information on effect sizes. For example, selective pressure means that deleterious alleles with strong effects are removed from the population more quickly and so may justify a prior that is more concentrated around zero. Functional annotations have been used to improve fine-mapping and genomic prediction \citep{weissbrod2020functionally, weissbrod2022leveraging} and could be incorporated to refine branch relatedness calculations for trait-based analyses.

% - Efficient branch GRM-vector products 
This paper also introduces new algorithms to efficiently compute branch relatedness. 
%JK I wouldn't talk about computing the full GRM at all, it's a 
% naive approach. Maybe we could works these pedigree GRM refs
% somehwere else?
Our approach to computing the branch GRM is similar to the algorithm for pedigree GRM \citep{emik1949systematic, cruden1949computation, henderson1976simple} in that the algorithm traverses the tree sequence to calculate the GRM for the chosen samples. Additionally, we develop a method for performing matrix-vector product calculations with the branch GRM without storing the full matrix in memory.  These algorithms leverage the sequential nature of the tree sequence data structure to reduce the required number of operations. \todo{\@Gregor - is this right? I've taken this from the old discussion but I'm not sure which bit of the pedigree computations are sequential so I may have misunderstood} This is akin to similar algorithms developed for pedigrees \citep{colleau2002indirect}. Such matrix-vector product calculations facilitate the efficient implementation of common analyses based on ARG relatedness, including principal component analyses.

Our branch GRM is equivalent to the eGRM put forward by \cite{fan2022genealogical} and the ARG-GRM put forward by \cite{zhang2023biobank}. The `e' in eGRM denotes `expected', reflecting that it represents the expectation of the canonical (site) GRM under a Poisson mutation process along ARG branches. We adopt the term \textit{branch} to highlight that this measure of similarity is derived from the extent of shared branch area between individuals, explicitly distinguishing it from genotype (site) GRMs.

% JK: again, let's just stay away from our implementation of 
% computing the full matrix. It's not interesting, and waters
% down the real contribution of a logarithmic-time algorithm
% for doing the product. The whole point should be that computing
% the full matrix is not something you would actually do.
Our approach to computing the branch GRM is similar to that of \cite{fan2022genealogical}, relying on traversing all the trees in the tree sequence to compute the GRM. By implementing our method directly in C in tskit, we achieve a moderate reduction in computational time compared to the eGRM package. In contrast, \cite{zhang2023biobank}, approximate the branch GRM by sampling new mutations on the branches and computing the standard GRM from these mutations
\citep{vanraden2008efficient, yang2010common}. While this sampling approach converges to the branch GRM under realistic mutation rates and avoids the need to update $n^2$ GRM elements for each branch, it does not compute the GRM exactly. Our method, in contrast, provides an exact computation of the branch GRM. Additionally, by implementing efficient matrix-vector product algorithms, we enable branch PCA on tree sequences with over one million samples in approximately 30 secondsâ€”far exceeding the computational feasibility of direct PCA on the branch GRM, which is limited to around 10,000 samples \citep{fan2022genealogical}.

\todo[inline]{A paragraph on computational scaling}
% TODO: mention matvec with pedigree and genotype GRMs
% \cite{colleau2002indirect} described how to obtain pedigree GRM
% for a set of individuals from a pedigree without explicitly calculating the full GRM.
% %
% This was achieved by working with the sparse inverse of the pedigree GRM and
% solving a system of equations for the susbset GRM.

\todo[inline]{TODO: If we touch on IBD, could mention \cite{huang2024estimating}
have an efficient representation of IBD that is ~linear?
Estimating evolutionary and demographic parameters via ARG-derived IBD --> maybe that work could be scaled by our algorithm!?}

% Outlook paragraphs

While relatedness has numerous downstream applications, we expect this perspective to be particularly impactful in phenotypic analyses, such as accounting for population structure in genome-wide association studies and improving genomic prediction, including polygenic scoring.

By capturing old and recent ancestry,
branch GRM could help with quantitative genomic modellling across ancestry groups
and with this improve portability of genomic predictions across the groups
\citep{legarra2021correlation, durvasula2021negative, wang2022challenges, prive2022portability, schultz2022stability, yair2022population}

\todo[inline]{Miscellaneous bits leftover from initial discussion below}
Branch GRM captures old and recent ancestry, while genotype GRM captures less of the recent ancestry
\citep[e.g.]{fan2022genealogical, young2022discovering}, suggesting that there would be no need to combine genotype GRM with pedigree GRM in certain applications spanning a wide range of relatedness \citep[e.g.]{vanraden2008efficient, kemper2021phenotypic}.

\todo[inline]{TODO: Paragraph on tslmm}

\todo[inline]{Future work: work across multiple tree sequences to manage multiple chromosomes}

\peter{Omit this probably:}
A recurring theme when comparing different notions of relatedness
is the sort of information used:
% for instance, the introduction of common genotyping led to the phrase ``genotypes provide realised GRM, while the pedigree provides expected GRM''.
for instance, pedigree relatedness averages over genetic relatedness,
while branch relatedness averages over mutation within a given ARG
(which would itself be nested within the population pedigree, if that were known).
So, we saw that expected branch relatedness can be simply related
to pedigree relatedness, but there is substantial variation
in the amount of branch relatedness realized between any pair of relatives,
especially on single chromosomes.
(However, there is very little remaining variation of genetic relatedness
given branch relatedness due to the action of mutation \citep{ralph2019empirical}.)