
% Summary paragraph
Recent advances in ARG inference have generated significant interest in leveraging ARGs for genetic analyses.
%
In this paper, we examined the relationship between different definitions of genetic relatedness
in the common context of additive traits on an ARG,
especially the emergent notion of ``branch'' relatedness.
%
We also demonstrated how branch relatedness compares with pedigree relatedness in simulations through a pedigree of French-Canadians.
%
We then described an algorithm for branch GRM-vector product,
to bypass the fundamental problem of quadratic complexity of computing and storing GRMs.
%
This algorithm allowed us to use randomized linear algebra methods
for branch PCA using an ARG on a million samples in 30 seconds
and less than 4GB of RAM.
%
We close the discussion with an outlook for the use of these algorithms in population and quantitative genomic analyses.

% Paragraph on key points behind branch relatedness
The described branch relatedness unifies several notions of relatedness into one framework
by leveraging the ARG encoding of how sampled genomes relate to each other.
One thing that distinguishes these notions of relatedness
is which aspects of genetics and genealogy are unobserved or observed,
and which are averaged over or fixed.
For instance, pedigree relatedness averages over
recombinations within a known pedigree,
genotype relatedness averages over typed loci stored in a genotype matrix, 
while branch relatedness averages over mutations on
inherited genome segments in an ARG.
Our branch relatedness is conceptually equivalent to the eGRM from \cite{fan2022genealogical} and the ARG-GRM from \cite{zhang2023biobank},
although we omit the scaling by $(p(1-p))^\alpha$ used by both.
The ``e'' in eGRM denotes expectation of the genotype GRM under a Poisson mutation process along ARG branches. 
At the risk of further confusing terminology,
we adopted the term \textit{branch} to highlight that this measure of similarity is derived from the extent of shared branch area between individuals,
explicitly distinguishing it from expectations that are conditional
on other quantities -- for instance, expected covariance given a pedigree
or expected covariance given a collection of genotypes (but not their effect sizes).
We have seen that branch relatedness varies substantially
for relatives of a given degree, in line with the theory
on pedigree and genetic ancestors
\citep{chang1999recent, weir2006genetic, hill2011variation, thompson2013identity, garciacortes2013variance}.
Does mutation contribute a large degree of variability
in addition to the branch relatedness?
\citet{fan2022genealogical} computed the ``varGRM'' to describe this
(for general theory see \citet{ralph2019empirical});
and in general the answer is ``no'' --
randomness due to the mutation process adds little variability
beyond recombination, except to small segments of the genome.
% the SD is proportional to the sqrt of the number of shared mutations
% if you want to be more concrete.

% Relatedness from a trait model
We have chosen to interpret GRM in the context of a generative model of traits following 
the initial definition of relatedness \citep{wright1922coefficients}.
However, the worth of a given GRM is determined by how well it works in practice,
rather than its theoretical justification,
and applications have motivated a number of interpretations and adjustments \citep{speed2015relatedness}.
However, adjusting the trait model gives a natural setting in which
to suggest extensions and the corresponding GRM is then a by-product of these extensions.
Although our current definition of relatedness assumes equal prior effects across all loci, one could consider alternatives whereby we incorporate prior information on effect sizes. For example, selection reduces frequency of deleterious mutations with strong effects from the population and such mutations may justify a different prior;
this prior might depend on mutation age in a similar way that the GRMs often weigh alleles by a function of their frequency \citep{speed2015relatedness}.
Functional annotations have been used to improve fine-mapping and genomic prediction \citep[e.g.][]{MacLeod2016Exploiting, weissbrod2020functionally, weissbrod2022leveraging} and could be incorporated as prior information on mutation effects, which will refine branch relatedness calculations for trait-based analyses.

% Branch GRM algorithms
% This paper also introduces a new algorithm to efficiently compute branch GRM and
% more importantly algorithms to efficiently compute operations \emph{with} the branch GRM. 
% It is natural to first develop an algorithm to compute a GRM from the observed data,
% but this approach is fundamentally challenging due to the $O(n^2)$ time and space
% computational complexity.
% For example, the early recursive algorithm for pedigree GRM
% \citep{emik1949systematic, cruden1949computation} has been
% superseded by linear algebra operations directly on the pedigree graph
% \citep{henderson1976simple, colleau2002indirect}.
% Also, genotype GRM-vector products can be performed directly via
% genotype matrix-vector products \citep{colleau2017fast}.
% We used the Schieber-Vishkin
% algorithm \citep{Schieber1988On,knuth2011combinatorial} to efficiently
% find MRCAs and corresponding branch GRM among genomes in an ARG,
% but this approach does not scale to genomic datasets of interest today, let alone in the future.

% JK: again, let's just stay away from our implementation of 
% computing the full matrix. It's not interesting, and waters
% down the real contribution of a logarithmic-time algorithm
% for doing the product. The whole point should be that computing
% the full matrix is not something you would actually do.
% PR: okay, I'm commenting this out and writing a new paragraph below.
% Our approach to computing the branch GRM is similar to that of \cite{fan2022genealogical}, relying on traversing all the trees in the tree sequence to compute the GRM. By implementing our method directly in C in tskit, we achieve a moderate reduction in computational time compared to the eGRM package. In contrast, \cite{zhang2023biobank}, approximate the branch GRM by sampling new mutations on the branches and computing the standard GRM from these mutations \citep{vanraden2008efficient, yang2010common}. While this sampling approach converges to the branch GRM under realistic mutation rates and avoids the need to update $n^2$ GRM elements for each branch, it does not compute the GRM exactly. Our method, in contrast, provides an exact computation of the branch GRM. Additionally, by implementing efficient matrix-vector product algorithms, we enable branch PCA on tree sequences with over one million samples in approximately 30 secondsâ€”far exceeding the computational feasibility of direct PCA on the branch GRM, which is limited to around 10,000 samples \citep{fan2022genealogical}.

% Better algorithm
%
Computing a full GRM is inherently a quadratic operation and therefore not feasible on large sample sizes. It is possible, however, to calculate GRM-vector products at a substantially lower computational cost.
% Working with the branch GRM with tree-by-tree algorithms is challenging, because the contribution of a branch above $n$ samples affects $O(n^2)$ entries in the GRM.
% With $n_S$ samples and $n_T$ local trees, each of which has $O(n_S)$ edges, a na{\"i}ve algorithm is $O(n_S^3 n_T)$.
% Here we discuss how this can be reduced to $O(n_S^2 n_T)$,
% but this is still infeasible for large samples.
With $n_S$ samples and $n_T$ local trees, our branch GRM-vector product algorithm has complexity $O(n_S + n_T \log n_S)$. This relies on two insights: first, we use local trees to
efficiently encode the low-dimensional block structure
of the contribution to the GRM of a single local tree;
and second, we leverage the fact that most tree structure
is shared across many local trees in the ARG.
This removes the need for approximate methods
such as the Monte Carlo sampling of mutations on the ARG
used by \citet{zhang2023biobank}.
The method is most similar to \citet{zhu2024variance},
who uses iterative algorithms
to compute GRM-vector multiplication with the
genotype GRM from Monte Carlo-sampled mutations,
but the algorithm is not given, so a more detailed comparison
is not possible.
We also provide a highly efficient vector-GRM-vector product algorithm,
similar to the classic algorithm for large pedigrees \citep{colleau2002indirect},
using the generic framework of \citet{ralph2020efficiently}.

% Outlook paragraph
This work provides the definition of branch relatedness based on a concrete trait model,
algorithms to efficiently compute with the corresponding branch GRM for millions of genomes, and
well documented and thoroughly tested open-source \tskit{} implementation.
%
These contributions are further opening a possibility for mega-scale population genetics and
quantitative genetics.
%
The clear definition of branch relatedness
(based on the fundamental ARG encoding of sampled genomes, a trait model
extendable with additional biological prior information)
will enhance the analyses of diverse and admixed genomic datasets
that are challenged by many evolutionary processes and data availability
\citep[e.g.][]{MacLeod2014Effects, Martin2017Human, durvasula2021negative,
wang2022challenges, yair2022population, RosFreixedes2022Genomic}.
%
The efficient branch GRM-vector product algorithm will speed-up
analyses of population structure, genome-wide associations,
heritability, and genomic prediction.