\input{preamble}

\begin{document}

% --- BRIEF INTRO -------------------------------------------------------------

\paragraph{Relatedness is one of the central concepts in genetics.}
%
In its most general sense, genetic relatedness refers to the notion of
similarity between individuals' genomes.
%
These similarities are usually organised as a pairwise comparison of the
genomes within an individual and between individuals, or groups of
individuals.
%
As one of the central genetic concepts, relatedness is used in many
applications \citep{weir2006genetic, speed2015relatedness}.
%
For example, to describe genetic variation within and between individuals
and groups of individuals in population genetics
\citep{crow2009introduction, charlesworth2010elements},
to analyse phenotype covariation between close and distant relatives in
quantitative genetics \citep{falconer1996introduction, lynch1998genetics},
and to estimate genetic changes in phenotypic variation over time in
evolutionary genetics \citep{walsh2018evolution, arnold2023evolutionary}.
%
Over time, genetic relatedness has been defined according to
pedigree \citep{fisher1919correlation, wright1922coefficients},
genotype \citep{cotterman1940calculus, malecot1948mathematiques, malecot1969mathemathics},
phylogeny \citep{lynch1991methods},
coalescent \citep{slatkin1991inbreeding}, and
recently ancestral recombination graph (ARG)
\citep{tsambos2022efficient, fan2022genealogical, zhang2023biobank}.
%
There have been numerous efforts to define relatedness according to these
concepts, and then to estimate it from the available data.
%
Despite the importance of genetic relatedness, these multiple definitions
seem disjoint, even out of sync, which adds to the confusion about the
several meanings attached to various relatedness coefficients
\citep{jacquard1975inbreeding, templeton1994inbreeding,
lacy1995clarification, rousset2002inbreeding}.
%
Here, we aim to connect these definitions in the light of today's abundant
genomic data and the recent rapid development of effective methods to infer
ARGs from such data
\cite{speidel2019method, kelleher2019inferring, zhang2023biobank, deng2024robust}.
% TODO: reviews/opinions by harris2019database, harris2023using
%
Recently, \cite{schraiber2024unifying} have also drawn parallels between
these definitions to study the estimation of genome-wide associations in
presence of population structure and associated environmental confounding.
%
While genetic relatedness is a huge topic on its own, a lot of development
has been done in relation to quantitative genetics.
%
We leverage some of this quantitative genetics work and focus throughout
only on the additive genetic model of phenotype variation, the workhorse
of much of quantitative genetics
\citep{falconer1996introduction, lynch1998genetics}.
%
Note that this model captures additive allele effects as well as
parts of non-additive genotype effects
\cite[e.g.][]{hill2008data, hivert2021gene}.
%
We demonstrate our work with the calculation of pedigree-, genotype-,
\textcolor{red}{phylogeny-?, coalescent-?}, and ARG-based relatedness 
for simulated data from the French Canadian population of
\cite{andersontrocme2023genes}.
% TODO: Can we do "phylogeny"-based relatedness between villages?
%       This is effectively Fst / group relatedness stuff!?
%
An important aspect of this application are patterns of historic migration
and settlement shaped by geographic constraints and how these processes are
reflected by different genetic relatedness measures.

% LONG-TIME SPAN OF LITERATURE
\paragraph{As indicated, genetic relatedness has been defined
and studied over a long period of time.}
%
We summarise some of the historical developments to review and connect
the different notions of genetic relatedness.
%
Some of this summary is an interpretation of a long history
with a vast amount of literature.
%
Hence, it is impossible to provide an explicitly chronological summary
without using current perspectives that leverage later developments.
%
Also, much of this development has occurred in different research communities
with limited interaction, including varied terminology.
%
In the following we first summarise the notion of genetic relatedness
according to pedigree, followed by re-defintion according to genotype, and
estimation of relatedness from observed genotype data.
%
While pedigree-based relatedness is inherently limited by pedigree
completness and the lack of concrete DNA information, genotype-based
relatedness avoids these limitations, but it does not directly inform
about the processes that generated that DNA variation.
%
We then summarise the notion of genetic relatedness according
to tree-like representations of genetic processes, first in the form of
phylogeny for species or populations and then in the form of coalescent
trees for specific DNA loci.
%
Finally, we connects all these different notions through ARGs and ARG-based
relatedness.

\textcolor{red}{TODO: then we WHAT?. Should we put the GAP and AIM here to
indicate the structure of the intro and the rest of the paper?}

% --- PEDIGREE ----------------------------------------------------------------

\subsection{Pedigree-based relatedness}

% TODO: Cut out some of the details below and move to appendix and
%       mention key concepts and definitions only?
%       Will write first all the stuff I think we have to cover for relatedness
%       (and there are several interconnected strands that might not seem 
%       relevant, but I think they are if we want to take advantage of ARGs).
%       Then we will later decide on how to move onwards having all the
%       "dump" material at hand. I am also writting all this text to organise
%       all my thoughts.

% START OF FORMAL PEDIGREE RELATEDNESS WORK BY FISHER & WRIGHT
\paragraph{Genetic relatedness was first formalised by \cite{wright1922coefficients},
who introduced the coefficients of relationship and inbreeding in the
context of a pedigree.}
%
These definitions complemented the seminal work of \cite{fisher1919correlation}
that formalised observed patterns of phenotypic resemblance among relatives.
%
By formulating phenotype observations as a sum of effects across many genes 
and their alleles, as well as environmental effects,
\cite{fisher1919correlation} has connected seemingly incompatible
observations of discrete ``Mendelian'' traits (affected mostly by a few
genes) and continuous ``biometric'' traits (affected by many genes and
environment).
%
To this end, \cite{fisher1919correlation} has used a linear model:
%
\begin{equation}
  y_i = \mu + g_i + e_i,\label{eqn:fisher_pheno}
\end{equation}
%
where
$y_i$ is the phenotypic value of individual $i$,
$\mu$ is the population mean,
$g_i \sim \N\left(0, \sigma_g^2\right)$
is genetic value (deviation of the individual $i$ from $\mu$
due to its set of inherited alleles and their effects) with genetic
variance $\sigma_g^2$, and
$e_i \sim \N\left(0, \sigma_e^2\right)$
is environmental deviation with environmental variance $\sigma_e^2$.
%
While \cite{fisher1919correlation} worked with specific types of relatives
and additive and non-additive genetic variation,
\cite{wright1922coefficients} worked with general pedigrees,
but implicitly only with additive genetic variation.
%
\cite{hill1996sewall} gave perspective on Wright's publications on genetic
relatedness.

% WHAT IS A PEDIGREE
\paragraph{Pedigree is a directed acyclic graph (DAG).} In this DAG, nodes
represent individuals and edges represent direct connections between parents
and their progeny.
%
In the standard case, each progeny has two parents; hence, the pedigree DAG
is sparse.
%
As we progress from contemporary individuals backward in time, we span
generations of known ancestors until we reach pedigree founders.
%
On this traversal, we can observe some individuals that have two, one, or
no known parents.
%
Pedigrees are hence practically always incomplete; progressively so
backward in time.

% WRIGHT'S RELATIONSHIP COEFFICIENT
\paragraph{Based on his path analysis of phenotypic values for pedigree
members, \cite{wright1922coefficients} defined
\textit{relationship coefficient between individuals $i$ and $j$ as the
correlation between their genetic values}:}
%
\begin{align} \label{eqn:wright_relationship}
  R_{i,j} & = \Cor\left(g_i, g_j\right) \\
          & = \frac{\Cov\left(g_i, g_j\right)} 
                   {\sqrt{\Var\left(g_i\right) \Var\left(g_j\right)}} \notag.
\end{align}
%
Although \cite{wright1922coefficients} did not clarify assumptions,
combining \cite{wright1921systems} and \cite{wright1922coefficients}
we can interpret the approach as follows by
convolving variation due to DNA processes and variation due to
processes between DNA and phenotypes.
%
Genetic value of a diploid individual under additive model
is the sum of the values of its genomes:
$g_i = g_{i,1} + g_{i,2}$ with:
%
\begin{align} \label{eqn:g_ind_var}
  \Var\left(g_i\right) & = \Var\left(g_{i,1}\right) + 
                           \Var\left(g_{i,2}\right) +
                           2\Cov\left(g_{i,1},g_{i,2}\right) \\
                       & = \nicefrac{1}{2}\sigma_g^2 +
                           \nicefrac{1}{2}\sigma_g^2 +
                           2K_{(i,1),(i,2)}\nicefrac{1}{2}\sigma_g^2 \notag \\
                       & = \left(1 + F_i\right)\sigma_g^2. \notag
\end{align}
%
Here, $K_{(i,1),(i,2)} = F_i$ is a measure of ``similarity'' between the
two genomes of individual $i$.
%
This similarity value can be positive or negative because it is a covariance
coefficient.
%
We use notation $F_i$ following \cite{wright1922coefficients} and
notation $K_{i,j}$ to denote what will later be named as a kinship
coefficient following translations of
\cite{malecot1948mathematiques, malecot1969mathemathics}.
%
For a pedigree founder there is no information about their ancestors,
hence we assume $F_i=0$ and $\Var\left(g_i\right) = \sigma_g^2$.
%
Between individuals we then have:
%
\begin{equation} \label{eqn:g_pair_cov}
  \Cov\left(g_i, g_j\right)=2K_{i,j}\sigma_g^2,
\end{equation}
%
where $K_{i,j}$ is similarity between genomes of individual $i$ and
genomes of individual $j$, with four possibilities in diploids:
%
\begin{equation}
  K_{i,j} = \frac{1}{4}\left(
    K_{(i,1), (j,1)} +
    K_{(i,1), (j,2)} +
    K_{(i,2), (j,1)} +
    K_{(i,2), (j,2)}\right).\label{eqn:kinship_pair}
\end{equation}
%
This leads to:
%
\begin{equation}
  R_{i,j} = \frac{2K_{i,j}}
                 {\sqrt{\left(1 + F_i\right)\left(1 + F_j\right)}},
  \label{eqn:wright_relationship_K}
\end{equation}
%
which indicates why the $2K_{i,j}$ is sometimes referred to as
the \textit{numerator relationship coefficient} \citep{henderson1976simple}.

% WRIGHT'S INBREEDING COEFFICIENT
\paragraph{\cite{wright1922coefficients} defined
\textit{inbreeding coefficient of individual $i$ as the correlation
between values of its genomes (``uniting gametes'')}:}
%
\begin{align} \label{eqn:wright_inbreeding}
  \Cor\left(g_{i,1}, g_{i,2}\right) & = \frac{\Cov\left(g_{i,1}, g_{i,2}\right)}
                                             {\sqrt{\Var\left(g_{i,1}\right) \Var\left(g_{i,2}\right)}} \\
                                    & = \frac{K_{(i,1),(i,2)}\nicefrac{1}{2}\sigma_g^2}
                                             {\sqrt{\nicefrac{1}{2}\sigma_g^2 \nicefrac{1}{2}\sigma_g^2}} \notag \\
                                    & = K_{(i,1),(i,2)} \notag \\
                                    & = F_i. \notag
\end{align}
%
Hence, Wright's inbreeding coefficient $F_i$ is by definition bounded
between -1 and 1.
%
Since parents generate and pass gametes to progeny at random, 
the inbreeding coefficient of an individual $i$ is
half of the numerator relationship coefficient between its mother $m(i)$ and
father $f(i)$, $F_i=K_{(i,1),(i,2)}=K_{m(i),f(i)}$.

% EARLY PEDIGREE ALGORITHMS
\paragraph{\cite{wright1922coefficients} showed how to calculate
relatedness coefficients in a general pedigree by tracing all pedigree
lineages between relatives.}
%
\cite{emik1949systematic} and \cite{cruden1949computation} devised a
simple procedure to calculate these coefficients between all pairs of
pedigree members.
%
This procedure has been later formalised as a recursive algorithm
\citep{henderson1976simple}.
%
The algorithm fills in the symmetric matrix
$\bsymb{G}$ with $2K_{i,j}$ values (covariance coefficients).
%
The algorithm assumes that a pedigree is topologically sorted
and proceeds from pedigree founders towards their descendants forward in
time.
%
The algorithm iterates over elements of $\bsymb{G}$ using two
expressions.
%
Expression for diagonal elements is:
%
\begin{equation} \label{eqn:G_ind}
  \bsymb{G}_{i,i} = 1 + F_i,
\end{equation}
%
where
$F_i = \nicefrac{1}{2}\bsymb{G}_{m(i),f(i)}$ if parents are known
or $F_i = 0$ if one or both parents are unknown.
%
Expression for off-diagonal elements is:
%
\begin{equation} \label{eqn:G_pair}
  \bsymb{G}_{i,j} = \nicefrac{1}{2}\bsymb{G}_{i,m(j)} +
                    \nicefrac{1}{2}\bsymb{G}_{i,f(j)}.
\end{equation}
%
In the latter, we replace $\nicefrac{1}{2}\bsymb{G}_{i,p(j)}$ with $0$
for any missing parent of $j$ ($p(j)$).
%
The matrix $\bsymb{G}$ used to be referred to as the
\textit{numerator relationship matrix} (NRM) or
pedigree-based NRM (PNRM or PRM) \citep{henderson1976simple},
though nowadays the term \textit{numerator} is often dropped.
%
The Wright's \textit{relationship matrix} with $R_{i,j}$ values
(correlation coefficients) is then conversion of covariance coefficient
matrix $\bsymb{G}$ to correlation matrix.
%
Matrices with $G_{i,j}$, $R_{i,j}$, or $K_{i,j}$ values are
often interchangeably called as a (genetic) relationship, relatedness,
or kinship matrix, confusing the distinction between different
coefficients.

\textcolor{red}{TODO: WHERE DO WE SHOW AN EXAMPLE? The example should work
across pedigree, genotypes, phylogeny, and genealogy and show relatedness
between individuals, inbreeding, and some group relatedness too, possibly
with some mutations and one recombination (we might need more than one rec
if we go back into the past, while being small, doh) - I would also like
to add an unknown parent and parents coming from different populations,
which is where ARGs would start to shine!}

% STAT/QUANT GENETICS WITH PEDIGREES
\paragraph{The work of Fisher and Wright has laid foundations for
statistical modelling in quantitative genetics, particularly through the
use of linear mixed models
\citep{falconer1996introduction, henderson1984applications,
lynch1998genetics, mrode2023linear}.}
%
Work with these type of models was initiated in 1950s
\citep{henderson1959estimation, robinson1991that, gianola2015one}.
%
Linear mixed model generalises Fisher's phenotype model to:
%
\begin{equation} \label{eqn:fisher_pheno_mat}
  \bsymb{y} = \bsymb{Xb} + \bsymb{Zg} + \bsymb{e},
\end{equation}
%
where
$\bsymb{y}$ is an $(n_y \times 1)$ vector of phenotypic values,
$\bsymb{X}$ is an $(n_y \times n_b)$ design matrix linking $\bsymb{y}$ with
$\bsymb{b}$ $(n_b \times 1)$ vector of fixed effects (such as population mean $\mu$),
$\bsymb{Z}$ is an $(n_y \times n_g)$ design matrix linking $\bsymb{y}$ with
$\bsymb{g}$ $(n_g \times 1)$ vector of random effects (such as genetic values), and
$\bsymb{e}$ is an $(n_y \times 1)$ vector of random residuals (such as environmental effects).
%
Further,
%
\begin{equation} \label{eqn:g_normal}
  \bsymb{g} \sim \N\left(\bsymb{0}, \bsymb{G}\sigma_g^2\right),
\end{equation}
%
where
$\bsymb{G}$ is \textit{genetic covariance coefficient matrix}
(such as Wright's pedigree-based NRM) and
$\sigma_g^2$ is genetic variance parameter, and
$\bsymb{e} \sim \N\left(\bsymb{0}, \bsymb{E}\sigma_e^2\right)$,
where
$\bsymb{E}$ is environmental covariance coefficient matrix and
$\sigma_e^2$ environmental variance parameter.
%
Classic literature used notation $\bsymb{A}$ instead of $\bsymb{G}$
\citep{henderson1984applications},
while modern literature uses notation $\bsymb{A}$ for pedigree-based NRM
and $\bsymb{G}$ for genotype-based NRM as we will show later
\citep[e.g.][]{vanraden2008efficient, mrode2023linear}.
%
We use notation $\bsymb{G}$ throughout and where we need to distingish
between different $\bsymb{G}$, we use a supscript, such as $\bsymb{G}^{x}$,
where $x$ denotes a source of information or a method.

% MORE PEDIGREE ALGORITHMS
\paragraph{Algorithms for pedigree-based numerator relationship matrix and
corresponding linear mixed models leverage pedigree DAG.}
%
Wright's pedigree model of a progeny genetic value given its parent's
genetic values is:
%
\begin{equation} \label{eqn:wright_geno}
  g_i = \nicefrac{1}{2}g_{m(i)} + \nicefrac{1}{2}g_{f(i)} + r_i,
\end{equation}
%
where
$\nicefrac{1}{2}$ denotes that each parent passes half of its genome to
progeny and 
$r_i$ is the Mendelian sampling deviation of $g_i$ from the expected
parent average $\nicefrac{1}{2}\left(g_{m(i)} + g_{f(i)}\right)$.
%
The Mendelian sampling deviation is generated by the random process of 
passing parental genomes to progeny through recombination, segregation,
and mutation.
%
\cite{henderson1976simple} has toplogically sorted pedigree, set $g_i = r_i$
for pedigree founders, and stacked all the individual pedigree models into
a system of equations $\bsymb{g} = \bsymb{Tr}$,
where $\bsymb{T}$ is an $(n_g \times n_g)$ lower-triangular design
matrix that recursively links genetic value of each individual $(g_i)$ to
Mendelian sampling deviations $(\bsymb{r})$ of the individual and
all of its ancestors - with values $1.00$ for individual,
$0.50$ for parents, $0.25$ for grandparents, etc., and
$0.00$ for non-ancestors.
%
% TODO: Consider pulling this TRT^T stuff out into work with Hanbin?
%       It is important to point here out the Mendelian sampling variance though
%       because pedigrees to give this information but in variance form
%       while most people just focus on expectations from pedigree.
Then we have
$\Var\left(\bsymb{g}\right) = \Var\left(\bsymb{Tr}\right)
  = \bsymb{TRT}^T\sigma_g^2=\bsymb{G}\sigma_g^2$,
where $\bsymb{R}$ is diagonal variance coefficient matrix for
Mendelian sampling deviations (=coefficients of conditional variance for
$g_i$ given the genetic values of parents $g_{m(i)}$ and $g_{f(i)}$ and
pedigree).
%
Diagonal elements of $\bsymb{R}$ are $1$ for pedigree founders and
$1 - \nicefrac{1}{4}\left(1 + F_{m(i)}\right)
   - \nicefrac{1}{4}\left(1 + F_{f(i)}\right)$ for non-pedigree  founders
($0.5$ for progeny with non-inbred parents and $0.0$ for progeny with
fully inbred parents).
%
The pedigree DAG and associated recursive set of individual pedigree models
are the basis for pedigree-based algorithms
\citep{emik1949systematic, cruden1949computation,
henderson1976simple, quaas1976computing,
thompson1979sire, quaas1988additive,
meuwissen1992computing, tier1999computing, colleau2002indirect, sargolzaei2005fast,
mcpeek2004best, gengler2007simple, garciacortes2010fine, aguilar2011efficient,
stranden2020bpop, nilforooshan2021alternative}.
%
For additional implicit genetic properties of the pedigree-based
$\bsymb{G}$ and the pedigree-based linear mixed model, see
\cite{kennedy1988genetic}.
%
One important property in relation to different relatedness matrices used
in linear mixed models, is that in the pedigree-based mixed model, the
genetic variance parameter $\sigma_g^2$ corresponds to the variance of
genetic values for the pedigree founders (base population);
$\bsymb{g}_f \sim \N\left(\bsymb{0}, \bsymb{I}\sigma_g^2\right)$
\citep{kennedy1988genetic, powell2010reconciling, legarra2016comparing}.

% PEDIGREE RELATEDNESS EXTENSIONS
\paragraph{There are many extensions of standard pedigree-based relatedness.}
%
Some of the extensions related to additive genetic modelling include:
%
group relatedness \citep{wright1949genetical, jacquard1975inbreeding, cockerham1976group},
%
``gametic'' relatedness \citep{smith1985efficient, schaeffer1989inverse},
% TODO: shouldn't we cite some even more classic papers here for gametic PRM?
%       Hasn't anyone done it before the 80s or have everyone aggregated gametic PRM into
%       individual PRM?
%
X-chromosome or haplo-diploid relatedness \citep{grossman1989inbreeding, fernando1990genetic},
%
uniparental lineages such as mitochondria \citep[e.g.][]{gibson1997cytoplasmic}
%
expected mutations \citep{wray1990accounting, casellas2008within},
%
% dominance \citep{there is a lot of work here} % skipping to focus just on additive genetic model(s)
%
imprinting \citep{schaeffer1989inverse, tier1993analysing},
%
% epistasis \citep{there is a lot of work here} % skipping to focus just on additive genetic model(s)
%
repeated rounds of selfing \citep{sneller1994sas, oakey2006joint},
%
``sparse'' versus ``dense'' plant breeding pedigrees \citep{kim2016numericware},
%
polyploidy \citep{kerr2012use, hamilton2018computation}, and
%
multiple populations with admixture \citep{vanraden1992accounting, garciacortes2006multibreed}.
%
There are further extensions in relation to using the pedigree-based model
\eqref{eqn:wright_geno} with the phenotype model \eqref{eqn:fisher_pheno_mat}
\citep{henderson1984applications, mrode2023linear}.

% GAMETIC RELATEDNESS
\paragraph{Gametic relatedness is defined between the genomes of individuals
\citep{smith1985efficient, schaeffer1989inverse} and is related to
kinship coefficient defined in the next sub-section.}
%
This is the most fundamental relatedness measure, because it
describes similarity between the units of inheritance - the individuals'
(haploid) genomes passed between generations.
%
For individual $i$ we have $g_{i,1}$ and $g_{i,2}$ with:
%
\begin{align} \label{eqn:wright_gamete}
    g_{i,1} & = \nicefrac{1}{2}g_{m(i),1} + \nicefrac{1}{2}g_{m(i),2} + r_{i,1},\\
    g_{i,2} & = \nicefrac{1}{2}g_{f(i),1} + \nicefrac{1}{2}g_{f(i),2} + r_{i,2}. \notag
\end{align}
%
Pedigrees do not give information about individual's genomes, hence gametic
relatedness was based on the probabilistic treatment of relatedness between
individual's genomes.
%
Gametic relatedness $\bsymb{G}^{g}$ can be aggregated to individual
relatedness $\bsymb{G}^{i}$ using:
%
\begin{equation} \label{eqn:G_ped_ind_from_gam}
  \bsymb{G}^{i} = \frac{1}{2}\bsymb{K} \bsymb{G}^{g} \bsymb{K}^T,
\end{equation}
%
where
$\bsymb{K} = \bsymb{I}_n \otimes [1\;1]$,
$\bsymb{I}_n$ is identity matrix for $n$ individuals, and
$\otimes$ is the direct (Kronecker) product of two matrices
\citep{smith1985efficient}.

% GROUP RELATEDNESS
\paragraph{Group relatedness is defined between groups of individuals
\citep{wright1949genetical, jacquard1975inbreeding, cockerham1976group}.}
%
While gametic relatedness measures similarity between pairs of genomes and
individual relatedness measures similarity between pairs of
individuals over their genomes, group relatedness measures similarity
between pairs of groups of individuals over their genomes.
%
Depending on the application, group relatedness is defined as the average
relatedness by sampling individuals or by sampling genomes.
%
We return to this topic in the next sub-section when mentioning population
level relatedness and associated F-statistics initiated by
\cite{wright1949genetical}.

% LIMITATIONS OF PEDIGREE INFORMATION
\paragraph{However, pedigree information is fundamentally limited for
describing exact genetic relatedness between individuals.}
%
Specifically, there are three fundamental limitations.
%
First, while pedigree founders are assumed unrelated due to lack of
past pedigree records, they always have some ``background''
relatedness by sharing now unknown ancestors.
%
Hence, pedigrees can only quantify the relatedness of individuals relative
to the background relatedness of the pedigree founders
\citep{wright1965interpretation, jacquard1975inbreeding, kennedy1988genetic,
powell2010reconciling, legarra2015ancestral, legarra2016comparing}.
%
In pedigree-based linear mixed models, the background relatedness is
implicitly captured by the genetic variance parameter ($\sigma_g^2$)
estimated from co-variation of phenotypic values.
%
That is, higher (lower) background relatedness reduces (increases) the
estimate of $\sigma_g^2$.
%
Second, pedigrees are practically always incomplete; increasingly so
backward in time.
%
Hence, we invariably have a variable depth of pedigree information for
most individuals, which challenges comparisons of pedigree-based
relatedness between individuals.
%
In the pedigree-based linear mixed model literature, the incompletness is
addressed using 
average relatedness by year of birth
\citep{vanraden1992accounting, aguilar2008recursive},
unknown parent groups,
\citep{thompson1979sire, quaas1988additive, muff2019animal},
and metafounders
\citep{legarra2015ancestral, garciabaccino2017metafounders}.
%
While complete and deep pedigrees are desirable and increase the accuracy
of relatedness estimates, they ultimately lead to high relatedness
coefficients among all living individuals since these all share ancestors
far back in time
\citep{jacquard1974genetic, jacquard1975inbreeding}.
%
Specifically, for recombining DNA under random mating, most recent common
ancestor (MRCA) of all $n$ current individuals is expected to have lived
on the order of $\log_2(n)$ generations ago with a small variance
\citep{chang1999recent, derida2000on}.
%
This age of MRCA is very recent even for large populations
($\log_2(10^6) \sim 20$) and much smaller compared to the expected age of
MRCA for a non-recombining DNA (such as mitochondria or a small haplotype)
of about $2n$ generations ago with a variance on the order of $n^2$,
assuming the coalescent model \citep{kingman1982on, chang1999recent, hein2004gene}.
%
% TODO: What all ancestors? MRCA's for all local trees (and the above is for one local tree)
Furthemore, all ancestors of current individuals are expected to have lived
only a bit further back in time of about $1.77 \log_2(n)$ generations ago
\citep{chang1999recent}.
%
Population structure and migration don't change these results qualitatively
\citep{hein2004pedigree, rohde2004modelling}.
%
Importantly, these results relate to pedigree ancestors and not to
actual DNA sharing between the ancestors and the current individuals.
%
Third, pedigree measures expected genetic relatedness, that is,
genetic variation in line with expectations and (co-)variances
according to the pedigree model.
%
Specifically, pedigree captures genetic variation between families;
in non-inbred families all pairs of full-sibs have a numerator
relationship coefficient of 0.50, half-sibs 0.25, etc.
%
Pedigree cannot explicitly capture genetic variation within families
caused by recombination, segregation, and mutation
\citep[e.g.,][]{visscher2006assumption, hill2011variation, thompson2013identity,
garciacortes2013variance, white2020effect}.
%
Instead, pedigree model posits how much genetic variation we expect
between families and within families (given the inbreeding coefficients
of parents as shown in the previous paragraph on conditional variance).
%
In the context of pedigree-based linear mixed models, this prior
enables us to estimate the genetic values and its two components,
parent average and Mendelian sampling deviation.
%
However, Mendelian sampling term are estimable only for individuals
with sufficient amount of own or relatives' phenotypic information
\citep{henderson1984applications, mrode2023linear}.
%
% These estimates are all trait-dependent, as is posterior covariance
% between them $\Var\left(\bsymb{g} | \bsymb{y}\right)$, which can be
% interpreted as trait specific genetic relatedness.
% TODO: Is this last part of the sentence correct?
%
%       Var(g | y) also includes estimation error, so this is then not
%       trait-specific relatedness as opposed to Var(g)=G\sigma_g^2 with
%       G being trait-agnostic (neutral) relatedness!?
%
%       gianola2020inferring infers p(G(\beta) | y) as trait-specific
%       genetic relatedness (in genomic era) and made me wonder how 
%       p(G(\beta) | y) and Var(g | y) are related!

% --- GENOTYPE - early-years - definitions and still probs on pedigrees -------

\subsection{Genotype-based relatedness - pre-genomic era}

% START OF FORMAL GENOTYPE RELATEDNESS WORK BY MALECOT - BUT FIRST DEFINE GENOS & ALLELES
\paragraph{With the discovery of the structure and role of DNA,
relatedness was redefined with respect to genotypes of individuals.}
%
Before we proceed, we will define genotype as a random variable and 
state its statistical properties.
%
Let $w_{i,l}$ represent genotype of individual $i$ at locus $l$ and
$w_{i,l,o}$ represent its $o$-th allele ($o$ denotes origin).
%
Assume a bi-allelic locus and encode alleles as $0$ (reference) and
$1$ (alternative), with an arbitrary choice of the reference allele.
%
Encode diploid genotype as the number (dosage) of alternative alleles:
%
\begin{equation} \label{eqn:dosage}
  w_{i,l} = w_{i,l,1} + w_{i,l,2},
\end{equation}
%
giving values $0$, $1$, or $2$.
%
Multi-locus alelles inherited from one parent (=haplotype) is then
a row-vector $\bsymb{w}_{i,o}$ of $0$s and $1$s, with $k$ indexing
the parent.
%
The corresponding multi-locus genotype is
a row-vector $\bsymb{w}_i$ of $0$s, $1$s, and $2$s.
%
Multi-allelic loci can be accommodated through the bi-allelic encoding by
expanding the number of elements in $\bsymb{w}_{i,o}$ and $\bsymb{w}_i$
per locus - equivalent to the linear model ``dummy'' encoding of a factor
with more than two levels.
%
The reference allele frequency in the population is $p_l$,
which is also the expected value:
%
\begin{equation} \label{eqn:allele_freq}
  \E\left(w_{i,l,o}\right)=p_l,
\end{equation} 
%
and the alternative allele frequency is $q_l = 1 - p_l$.
%
Assuming random pairing of alleles under Hardy-Weinberg equilibrium
(=binomial sampling of alleles), genotype frequencies are:
%
\begin{equation} \label{eqn:geno_prob_hwe}
  \Pr\left(w_{i,l} = \left[2, 1, 0\right]\right) =
    \left[p^2_l, 2 p_l q_l, q^2_l\right].
\end{equation}
%
Non-random pairing of alleles can be parameterised with additional
parameter $F_l \in [-1, 1]$, inducing correlation between underlying
Bernoulli draws of alleles
\citep{wright1921systems, wright1922coefficients, wright1949genetical}:
%
\begin{equation} \label{eqn:geno_prob_f}
  \Pr\left(w_{i,l} = \left[2, 1, 0\right]\right) =
    \left[p^2_l + p_l q_l F_l, 2 p_l q_l(1 - F_l), q^2_l + p_l q_l F_l\right],
\end{equation}
%
with a constraint on possible values for $p_l$ and $F_l$ to ensure
a proper genotype distribution \citep{bahadur1961representation}.
%
In addition to inbreeding, population structure also impacts distribution
of genotypes \citep{wahlund1928zusammensetzung}, which we touch upon later.
%
Expectation for diploid genotype is:
%
\begin{equation} \label{eqn:geno_exp}
  \E\left(w_{i,l}\right) = 2p_l.
\end{equation}
%
% = [[0 * (q^2 + q p F)] +
%    [1 * 2q p(1 - F)] +
%    [2 * (p^2 + q p F)]]
% = [[0] +
%    [2q p - 2q p F] +
%    [2p^2 + 2q p F]]
% = 2p^2 + 2q p
% = 2p^2 + 2(1 - p) p
% = 2p^2 + 2p - 2p^2
% = 2p
%
% This assumes just drift (some hets go into both homs) and no selection;
% or maybe this can capture combined drift and selection. Anyway, if there
% would be selection, then the expectation could change with time. But, we can not
% parameterise three probabilites with more than two parameters (now we have p and F).
% Would need to add a fitness/selection coefficient, which is relevant across
% generations, not within a generation (here we are not talking about a time component).
%
Variance is then:
%
\begin{align} \label{eqn:geno_var}
  \Var\left(w_{i,l,o}\right) & = p_l q_l,\\
  \Var\left(w_{i,l}\right)   & = 2 p_l q_l (1 + F_l). \notag
\end{align} 
%
% = E(w^2_{i,l}) - E(w_{i,l})^2 =
% = [[0^2 * (q^2 + q p F)] +
%    [1^2 * (2q p(1 - F))] +
%    [2^2 * (p^2 + q p F)]] - (2p)^2
% = [[0] + [2q p(1 - F)] + [4p^2 + 4q p F]] - [4p^2]
% = 2q p(1 - F) + 4q p F
% = 2q p(1 - F + 2F)
% = 2q p(1 + F) # with F=[-1,0,1/2,1] this is [0,2qp,3qp,4qp] so all looks good
%
Covariance between two randomly sampled alleles is:
%
\begin{align} \label{eqn:allele_cov}
  \Cov\left(w_{i,l,o}, w_{j,l,o}\right) & = p_l q_l F_l \\
% = E(w_{i,l,o} w_{j,l,o}) - E(w_{i,l,o}) E(w_{j,l,o}) =
% = [[0*0 * (q^2 + q p F)] +
%    [0*1 * (2q p(1 - F))] +
%    [1*1 * (p^2 + q p F)]] - p p
% = [[0] + [0] + [p^2 + q p F]] - p^2
% = q p F # with F=[-1,0,1/2,1] this is [-qp,0,1/2qp,qp] so all looks good
%
                                        & = p_l q_l \Cor\left(w_{i,l,o}, w_{j,l,o}\right). \notag
\end{align}
% Cor(w_{i,l,o}, w_{j,l,o}) = Cov(w_{i,l,o}, w_{j,l,o}) / sqrt(Var(w_{i,l,o}) Var(w_{j,l,o}))
%                           = Cov(w_{i,l,o}, w_{j,l,o}) / p q
% Cov(w_{i,l,o}, w_{j,l,o}) = p q Cor(w_{i,l,o}, w_{j,l,o})
%                           = p q F
%                           = p_l q_l K_{(i,l,o),(j,l,o)}$.
% Noting that correlation is constrained by p to give a proper genotype distribution.
%
Covariance between two randomly sampled genotypes is:
%
\begin{align} \label{eqn:geno_cov}
  \Cov\left(w_{i,l}, w_{j,l}\right) & = 4 p_l q_l K_{i,j,l} \\
% = E(w_{i,l} w_{j,l}) - E(w_{i,l}) E(w_{j,l}) =
% = [[0*0 * (q^2 + q p F_i) (q^2 + q p F_j) K] +
%    [0*1 * (q^2 + q p F_i) (2q p(1 - F_j)) K] +
%    [0*2 * (q^2 + q p F_i) (p^2 + q p F_j) K] +
%    [1*0 * (2q p(1 - F_i)) (q^2 + q p F_j) K] +
%    [1*1 * (2q p(1 - F_i)) (2q p(1 - F_j)) K] +
%    [1*2 * (2q p(1 - F_i)) (p^2 + q p F_j) K] +
%    [2*0 * (p^2 + q p F_i) (q^2 + q p F_j) K] +
%    [2*1 * (p^2 + q p F_i) (2q p(1 - F_j)) K] +
%    [2*2 * (p^2 + q p F_i) (p^2 + q p F_j) K]] - (2p)^2
% = [[0] +
%    [0] +
%    [0] +
%    [0] +
%    [(2q p(1 - F_i)) (2q p(1 - F_j)) K] +
%    [2 (2q p(1 - F_i)) (p^2 + q p F_j) K] +
%    [0] +
%    [(p^2 + q p F_i) (2q p(1 - F_j)) K] +
%    [4(p^2 + q p F_i) (p^2 + q p F_j) K]] - 4p^2
% = [[(2q p(1 - F_i)) (2q p(1 - F_j))] +
%    [2 (2q p(1 - F_i)) (p^2 + q p F_j)] +
%    [(p^2 + q p F_i) (2q p(1 - F_j))] + 
%    [4(p^2 + q p F_i) (p^2 + q p F_j)]]K - 4p^2
% assuming F_i = F_j = F
% = [[(2q p(1 - F))^2] +
%    [2 (2q p(1 - F)) (p^2 + q p F)] +
%    [(p^2 + q p F) (2q p(1 - F))] +
%    [4(p^2 + q p F) (p^2 + q p F)]]K - 4p^2
% = [[4q^2 p^2(1 - F_i)^2] +
%    [2 (p^2 + q p F) (2q p(1 - F))] +
%    [  (p^2 + q p F) (2q p(1 - F))] +
%    [4 (p^2 + q p F) (p^2 + q p F)]]K - 4p^2
% --> Tried Wolfram Alpha but ran out of time with the free version
% --> Should be able to deduce this result!
%     If we deal with the same individual, then Cov()=Var()= 2q p(1 + F) = 2 p q (assuming no inbreeding)
%     if we deal with different individuals that are not related in any way (or have average relatedness?!), then Cov()=0
%     if we deal with a the same individual that is fully inbred then we should get 4pq=2pq(1+1)
% --> Weir, but also Astle and Balding have 4pqK, here K is [-1,1]
%     with K=[-1,0,1/2,1] this is [-4pq,0,2pq,4qp] so all looks good
                                    & = 4 p_l q_l \Cor\left(w_{i,l}, w_{j,l}\right). \notag
\end{align}
% Cor(w_{i,l}, w_{j,l}) = Cov(w_{i,l}, w_{j,l}) / sqrt(Var(w_{i,l}) Var(w_{j,l}))
%                       = Cov(w_{i,l}, w_{j,l}) / sqrt(2 p q (1 + F_i) 2 p q (1 + F_j))
%                       = Cov(w_{i,l}, w_{j,l}) / 2 p q sqrt((1 + F_i) (1 + F_j))
%                       = Cov(w_{i,l}, w_{j,l}) / 2 p q sqrt(1 + F_j + F_i + F_iF_j)
% assuming F_i = F_j = F --> 1 + 2F + F^2 = (1 + F)^2 --> sqrt(...) = (1 + F)
% (so, both i and j must come from the same (sub-)population and have alleles sampled at random
%  from that population so the population F applies to both)
%                       = Cov(w_{i,l}, w_{j,l}) / 2 p q (1 + F)
% Cov(w_{i,l}, w_{j,l}) = 2 p q (1 + F) Cor(w_{i,l}, w_{j,l})
%                       = 2 p q (1 + F) K
% TODO: Why is this not 4pqK?

% START OF FORMAL GENOTYPE RELATEDNESS WORK BY MALECOT
\paragraph{Seminal work on relatedness with respect to genotypes has been
done by \cite{cotterman1940calculus} and
\cite{malecot1948mathematiques, malecot1969mathemathics}.}
%
The work of Malecot is summarised in
\cite{nagylaki1989gustave} and \cite{slatkin2002modern}.
%
Although Malecot initially followed (and generalised) the covariance and
correlation work of Fisher and Wright, he has later probabilistically
treated genotype outcomes of recombination, segregation, and mutation
processes.
%
Specifically, he formalised the concept of ``genic'' random variables
for alleles of an individual $(w_{i,l,o})$ and associated probability of
identity between alleles.
%
With these definitions, Malecot separated the study of genotype and
allele variation from their effects on phenotypic values
(as described in the previous sub-section).
%
He distinguished
\textit{identity by state} (IBS; where two alleles are identical in their
DNA sequence) and
\textit{identity by descent} (IBD; where two alleles are identical due to
descending from the same ancestral DNA sequence).
%
With this retrospective view of allele's histories, Malecot anticipated
research on the coalescence of homologous DNA sequences; a topic we
cover later.
%
Two alleles can be IBS because they descended from the same origin
(hence, they are also IBD)
or because two different mutations descended from different origins
and created the same type of allele.
%
Conversely, two alleles can be IBD but differ in DNA sequence due to
mutation(s) (hence, they are not IBS).
%
The opportunity for mutation increases with the length of the DNA sequence
and time, which therefore impact the identity of alleles.
%
In applications, the amount and distribution of mutations can impact IBD
inference.
%
Hence practical definitions of IBD include various thresholds
with respect to the length of DNA sequence (associated with recombination
events) and the number of of mutations (associated with time to the most
recent common ancestor).
%
Choice of these thresholds imply a choice of the allele's origin
(defining a reference point/population), meaning that IBD is often
expressed relative to the chosen reference population.
%
Depending on the setting and definitions, this reference population
might or might not align with the pedigree founders.
%
\cite{browning2012identity}, \cite{thompson2013identity}, and
\cite{wakeley2016coalescent} provide recent reviews on IBD.

% INBREEDING & KINSHIP AS PROBABILITIES OF IBD
\paragraph{Malecot redefined Wright's work based on the
probability of identity of alleles; representing the
notion of random sampling of alleles from individuals.}
%
He defined \textit{inbreeding (=``consaguinity'') coefficient of
individual $i$ as the probability that the two alleles of a diploid
individual are IBD}:
%
\begin{equation} \label{eqn:malecot_inbreeding}
  F_i = \Pr\left(I\left(w_{i,l,1} \equiv w_{i,l,2}\right)\right),
\end{equation}
%
where probability is considered over an evolutionary process or for
practical purposes over loci along the genome, assuming the loci are
non-independent realisations of the evolutionary process.
%
Because the alleles of an individual have descended from its parents
through the random inheritance process, the probability that its alleles
are IBD is equivalent to the probability that its parent's alleles are IBD.
%
In diploids, we have four possible combinations when drawing an allele
from each parent at random with replacement.
%
To this end, Malecot defined \textit{kinship (=``parente'') coefficient
as the probability that two randomly sampled alleles from two individuals
($i$ and $j$) are IBD}:
%
\begin{equation} \label{eqn:malecot_kinship}
  K_{i,j} = \Pr\left(\frac{\sum_{o=1}^2 \sum_{o'=1}^2 I\left(w_{i,l,o} \equiv w_{j,l,o'}\right)}{4}\right),
\end{equation} 
%
where $o$ and $o'$ are indices for the origin of alleles of an individual.
%
When applied to one individual, kinship coefficient is:
%
\begin{equation} \label{eqn:kinship_ind}
  K_{i,i} =\nicefrac{1}{2}\left(1 + F_i\right),
\end{equation} 
%
which is half of Wright's variance coefficient for the genetic value
of a diploid individual $i$,
$\Var\left(g_i\right)=\left(1+F_i\right)\sigma_g^2$ \eqref{eqn:g_ind_var}.
The kinship coefficient is also called the \textit{coancestry} coefficient
\citep{falconer1996introduction}.
% 
The \textit{relationship matrix} and \textit{numerator relationship matrix}
are sometimes called the \textit{kinship matrix} due to their close
connection, blurring the distinction between probability, covariance, and
correlation coefficients
\citep{jacquard1975inbreeding, templeton1994inbreeding, lacy1995clarification, rousset2002inbreeding}.

% WHILE THESE COEFFICIENTS WERE DEFINED ON GENOS, THERE WAS NO GENO DATA, SO PEDIGREES WERE USED
\paragraph{These genotype-based definitions of relatedness were proposed
before DNA data was available.}
%
Malecot hence treated alleles and genotypes as unobserved random variables
and used pedigrees to calculate the probabilities of IBD.
%
His pedigree results were equivalent to Wright's, indicating the close
connection between the IBD probability and the correlation of alleles
\citep{malecot1948mathematiques, malecot1969mathemathics,
wright1965interpretation, thompson1976population, meulepas1991probabilistic,
slatkin2002modern, rousset2002inbreeding}.
%
His pedigree-based expressions are equivalent to the procedure of
\cite{emik1949systematic} and \cite{cruden1949computation} to obtain
numerator relationship coefficients (\ref{eqn:G_ind}, \ref{eqn:G_pair}),
appart from the factor $2$ in diploids:
%
\begin{align} \label{eqn:kinship_rules}
  K_{i,i} & =\nicefrac{1}{2}\left(1 + F_i\right), \\
  F_i & = K_{m(i),f(i)}, \notag \\
  K_{i,j} & = \nicefrac{1}{2}(K_{i,m(j)} + K_{i,f(j)}). \notag
\end{align} 
%
% TODO: GG is confused about the above - the literature refers to
%   equivalence between probability and correlation coefficients,
%   BUT we seem to have "equivalence" between probability and
%   covariance coefficients (not correlation coefficients)!?
%   Jacquard in his book addressed this situation by working with
%   probabilities regarding kinship and inbreeding, but when talking
%   about phenotypic resemblance he stated that he will use correlations:
%   "In this section, we shall study resemblance between related individuals
%   and their quantitative character. The closeness of the resemblance
%   between two individuals who are related in a particular way will be
%   measured by the correlation coefficient of the character"
%   --> Once cleared, we should revise the "blurring" parts across the "intro"!?

% POSITIVE AND NEGATIVE VALUES FOR THE COEFFICIENTS
\paragraph{While Wright's coefficients are based on covariance and
correlation and can hence be positive or negative,
Malecot's coefficients are based on probabilities and are hence
non-negative.}
%
This inconsistency is due to definition and interpretation of coefficients
relative to a reference population
with corresponding allele and genotype frequencies
\citep{wright1965interpretation, jacquard1975inbreeding,
slatkin2002modern, rousset2002inbreeding, powell2010reconciling}, 
but also due to estimation when working with genomic data as described later
\citep[e.g.][]{wang2014marker, weir2017unified, ackerman2017estimating}.
%
Wright interpreted his coefficients as follows
\citep{wright1949genetical, wright1965interpretation}.
%
When the reference population are ``distant'' individuals,
relatedness coefficients within a population of ``current''
individuals are positive.
%
The positive values indicate an increased homozygosity (IBS),
and with it increased autozygosity (IBD),
at the expense of heterozygosity in the current population,
relative to the distant reference population.
%
When the reference population are current individuals,
some coefficients can be negative.
%
The negative values indicate a decreased homozygosity and autozygosity
to the benefit of heterozygosity in the current population, relative
to the current reference population.
%
Negative coefficients are particularly common with ``outbreeding''
- mating across divergent populations.

% WRIGHT'S STATISTICS
\paragraph{The relative nature of relatedness coefficients is the basis
for describing relatedness across (sub-)populations that can be defined
temporally, geographically, or some other grouping.}
%
To this end, \cite{wright1949genetical} introduced a series of
F-statistics.
%
$F_{IT}$ to describe relatedness of individuals (I) with respect to
a distant/total/ancestral/etc. reference population (T).
%
$F_{ST}$ to describe relatedness of sub-populations (S) with respect to
a distant/total/ancestral/etc. reference population (T).
%
$F_{IS}$ to describe relatedness of individuals (I) with respect to
a current/sub-population/etc. reference population (S).
%
These $F$ coefficients describe average relatedness between individuals
with respect to their groups and as such describe group relatedness
by comparing alleles randomly sampled from the groups
\cite{wright1949genetical, jacquard1975inbreeding, cockerham1976group}.
%
The relationship between these coefficients is:
%
\begin{align} \label{eqn:fst}
      F_{IT} & = F_{IS} + (1 - F_{IS}) F_{ST}\\
  1 - F_{IT} & = \left(1 - F_{IS}\right)\left(1 - F_{ST}\right). \notag
\end{align}
%
This relationship can be expanded to
an arbitrary number of levels \citep{ochoa2021estimating} and
hierarchical (tree-like) population structure \citep{maryhuard2023fast}.
%
Reviews on the definitions, interpretation, and
estimation of F-statistics are available in
\cite{cockerham1969variance}, \cite{cockerham1973analyses},
\cite{holsinger2009genetics}, \cite{bhatia2013estimating},
\cite{garciabaccino2017metafounders}, \cite{ochoa2021estimating},
and \cite{maryhuard2023fast}.
%
There is another set of F-statistics frequently used in modern population
genetics - \textit{sensu} \cite{patterson2012ancient}.
%
These F-statistics are defined as covariances of genic variables
or as branch lengths in population trees \citep{peter2016admixture},
and have a close relationship with the popular Principal Component
Analysis (PCA) of allele and genotype variation \citep{peter2022geometric}.

% CONNECTION BETWEEN IBS AND IBD & F-STATS
\paragraph{The relative nature of relatedness coefficients is used to
connect IBS and IBD in specific settings.}
% 
Set pedigree founders as a reference population of nominally unrelated
individuals and track IBD relatedness from the founders forward in time,
meaning that $F_{IS}$ measures $F_{IBD}$ since the founders.
%
Then, define background relatedness of the founders as IBS due to their
shared history, meaning that $F_{ST}$ measures $F_{IBS}$.
%
This setting gives us
\citep[e.g.,][]{jacquard1975inbreeding, powell2010reconciling,
legarra2015ancestral, legarra2016comparing}:
%
\begin{align} \label{eqn:fibd_fibs}
      F_{IT} & = F_{IBD} + (1 - F_{IBD}) F_{IBS}\\
  1 - F_{IT} & = \left(1 - F_{IBD}\right)\left(1 - F_{IBS}\right). \notag
\end{align}
%
Expression \eqref{eqn:fibd_fibs} is often used to denote the probability
of observing two IBS alleles, say a homozygous genotype for alternative
allele in individual $i$ at locus $l$:
%
\begin{align} \label{eqn:fibd_fibs_genotype}
  \Pr\left(w_{i,l} = 2\right) & = F_{IBD,i} \Pr\left(w_{i,l,o} = 1\right) +
                                  (1 - F_{IBD,i}) \Pr\left(w_{i,l,o} = 1\right)^2,
\end{align}
%
where the first term is the probability of homozygosity due to IBD and the
second term is the probability of homozygosity due to IBS (assuming
Hardy-Weinberg equilibrium \eqref{eqn:geno_prob_hwe}).
%
Furthemore, the expected number of genotype matches between two 
% TODO: Have I deleted this bit or do I have to finish it?
%
% TODO: jacquard1974genetic has some nice examples in his book should we
%       want to show anything in this space of F_IT, F_IBS, & F_IBD (section 1.5.2)
% TODO: jacquard1974genetic also has a nice passage: 1.5.3 "Remote" and "absolute"
%       consaguinity, where he describes that pedigree gives us some information
%       but clearly this is a limited information ... what fraction of the total/
%       absolute relatedness are we measuring - this question does not have a meaning
%       since all probability questions relate to information at hand and associated
%       assumptions --> hence improved coefs by taking IBS into account should
%       be more correct, but this is simply just a shift in the reference point.
%       He then goes into saying that if we count how many ancestors we would need
%       to all be unrelated and compare how many people actually lived long time ago
%       we see that we all must be "very" related --> this is a nice opening/link
%       to get to coalescent thinking and mutations etc.
% TODO: Get back to this and revise IBS&IBD once when we touch on coalescent and
%       IBD in that context

% COEFFICIENTS BETWEEN GENOTYPES, INSTEAD OF JUST ITS ALLELES
\paragraph{While inbreeding and kinship coefficients capture information
about the identity of alleles between individuals, they do not fully
capture information about the identity of their genotypes.}
%
To measure the identity of genotypes between individuals,
\cite{harris1964genotypic} and \cite{gillois1965relation} introduced
a larger set of detailed identity coefficients.
%
There are 15 detailed identity coefficients for diploids.
%
However, depending on the application the detailed identity coefficients
can be condensed to:
9 if the paternal and maternal origin of alleles is not of interest,
7 if the two individuals have the same level of inbreeding, or
3 if the two individuals are not inbreed.
%
\cite{jacquard1972what} and \cite{jacquard1974genetic} comprehensively
describe this topic, while \cite{weir2006genetic} provides a recent review.
%
For the calculations of detailed identity coefficients
from pedigree data see
\cite{jacquard1966logique, cockerham1971higher, karigl1981recursive, garciacortes2015novel}
and from genomic data see
\cite{csuros2014nonidentifiability, garciacortes2014coefficient,
ackerman2017estimating, graffelman2024estimation}.
%
The detailed or condensed identity coefficients are required for 
determining type of relationship between pairs of individuals
\citep[e.g.][]{thompson1975estimation, weir2006genetic} or 
study of non-additive genetic variation
\citep{jacquard1974genetic, lynch1998genetics, slatkin2002modern, barton2023infinitesimal},
while inbreeding and kinship coefficients are sufficient for the study of
additive genetic variation - the focus of this work.

% --- GENOTYPE - based on actual marker data ----------------------------------

\subsection{Genotype-based relatedness - genomic era}

% EARLY GENOTYPE-BASED RELATEDNESS ESTIMATION WITH A FEW TO HUNDREDS OF MARKERS
\paragraph{Developments of molecular genetics enabled marker data
generation and estimation of genotype-based relatedness.}
%
Early work used bioassay markers (blood groups and allozymes)
to estimate group or individual relatedness coefficients
\citep[e.g.][]{li1953some, morton1971bioassay, thompson1975estimation, pamilo1982measuring, queller1989estimating}.
%
\cite{li1953some} used a method of moments to estimate population
inbreeding coefficient from genotypes at locus $l$ by comparing
the expected heterozygosity under Hardy-Weinberg equilibrium $H_{e,l} = 2 p_l q_l$ \eqref{eqn:geno_prob_hwe} with
the observed heterozygosity with inbreeding $H_{o,l} = 2 p_l q_l (1 - F_l)$ \eqref{eqn:geno_prob_f}
(of equivalently comparing the corresponding variances \eqref{eqn:geno_var}),
giving:
%
% TODO: how should we denote true value and an estimator? \hat{F}?
\begin{align} \label{eqn:LiHorvitz_f}
  F_l & = \nicefrac{\left(H_{e,l} - H_{o,l}\right)}{H_{e,l}}\\
      & = 1 - \nicefrac{H_{o,l}}{H_{e,l}}. \notag
\end{align}
%
% TODO: This F_l here is F_{ST}_l, right!? Should we denote it as such?
%       I didn't do it before when defining Var(w_{i,l}) or Cov(w_{i,l,o}, w_{j,l,o})
%       because I brought in F_{ST} after that point. Maybe we should now
%       make this connection?
%
Depending on the ratio $\nicefrac{H_{o,l}}{H_{e,l}}$, $F_l$ can be zero,
positive, or negative \citep{wright1921systems, wright1922coefficients, wright1949genetical, li1953some}.
%
Assume that a population is under Hardy-Weinberg equilibrium so that
$H_{o,l} = H_{e,l}$, which gives $F_l = 0$.
%
Now assume that a population consists of only homozygous individuals but
that $p > 0$; then $H_{o,l} = 0$ and $F_l = 1$.
%
Finally, assume that a population consists of only heterozygous individuals,
hence $p = 0.5$ and $H_{o,l} = 2H_{e,l}$, giving $F_l = 1$.
%
\cite{thompson1975estimation} on the other hand initated use of maximum
likelihood method to improve the estimation of relatedness coefficients
from data.
%
This early work had limited power due to a small number of markers, many
revealing only two types due to dominance.
%
Improvements followed from using several dozens to hundreds of DNA markers,
such as multi-allelic minisatelites and microsatellites, or
increasingly bi-allelic single nucleotide polymorphisms (SNP)
\citep[e.g.][]{ritland1996estimators, milligan2003maximum, weir2006genetic}.
%
As anticipated by \cite{thompson1975estimation}, these early studies
found that the number of markers is critical for the quality of estimates
and distingishing types of relationships between pairs of individuals
\citep{weir2006genetic}.
%
The large number of markers is required to minimize the ambiguity between
IBS (due to background relatedness) and IBD (due to recent relatedness) \eqref{eqn:fibd_fibs}.
%
This ambiguity led the development of relatedness estimators
that account for background relatedness and possible population
structure - parameterised by allele frequencies or F-statistics -
as driven by evolutionary history of the (sub-)populations
\citep{balding1994dna, anderson2007maximum, wang2011unbiased}.
%
\cite{rousset2002inbreeding} and \cite{wang2014marker} note that 
some of the unbiased estimators mentioned above assume null
average relatedness, meaning that some pairs of individuals will have
a negative estimate of relatedness, which is challenging to interpret
as the probability of IBD.
%
\cite{rousset2002inbreeding} interpreted these estimators as a ratio of
probabilities of IBS that approximate the ratio of probabilities of IBD.
%
He further noted that the resulting estimates measure how much higher
(or lower) is the probability of IBD for pairs of individuals to the
average probability for all pairs of individuals.

% VARIATION IN RELATEDNESS DUE TO RECOMBINATION - OVERALL AND ALONG GENOME
\paragraph{Increasing the number of markers improved accuracy, but also
revealed substantial variation from expected relatedness coefficients
between pairs of individuals - overall as well as along genome regions.}
%
Namely, recombination and segregation sample combinations of parental
chromosomes passed to progeny.
%
This random process of Mendelian sampling drives variation in actual
relatedness between individuals as well as variation in ancestry of
alleles along genome regions.
%
This substantial variation in relatedness overall and along the genome
was anticipated by the theoretical work on linkage-disequilibrium and
IBD of chromosome segments in finite populations
\citep{hill1968linkage, ohta1969linkage, sved1971linkage,
stam1980distribution, donnelly1983probability}.
%
Capturing the variation in relatedness by resolving recombination and
segregation of genomes within pedigrees enabled estimation of associations
between genome regions and phenotypes in linkage studies
\citep[e.g.][]{fernando1989marker, terwilliger1994handbook,
lander1995genetic, ott2015genetic}.
%
These studies use markers to infer IBD of chromosome segments
between pedigree members, while pedigree founders are assumed independent
beyond the background relatedness due to IBS
\citep[e.g.][]{elston1971general, lander1987construction,
daywilliams2011linkage, ott2015genetic, whalen2018hybrid}.
%
While linkage studies are powerfull for finding large effect loci,
identified genome regions are often broad due to a limited number of
captured recombinations within pedigrees.

% MODERN GENOTYPE-BASED RELATEDNESS ESTIMATION WITH MANY GENOME-WIDE MARKERS
\paragraph{Further developments in molecular genetics enabled genome-wide
genotyping with SNP arrays and increasingly whole-genome resequencing.}
%
With SNP arrays we capture genetic variation at thousands to
about a million of loci.
%
With whole-genome resequencing we capture variation at several millions
of loci.
%
While whole-genome resequencing has the potential to capture all variation,
some variant types (such as SNP) are more easily captured accurately than
others (such as structural variants).
%
The estimation of relatedness between populations and individuals based on
genome-wide genotype data has been a major focus of research, with
a multitude of proposed methods, software, and applications.
%
Some studies focused on estimating genomic relatedness
\citep[e.g.][]{leutenegger2003estimation, vanraden2007efficient,
manichaikul2010robust, powell2010reconciling,
toro2011note, wang2014marker, speed2015relatedness,
weir2017unified, ackerman2017estimating, weir2018how,
ochoa2021estimating, maryhuard2023fast}.
%
The estimated relatedness coefficients between pairs of individuals are
stored in a matrix $\bsymb{G}$, commonly referred to as
genotype-based/genome-based/genomic relatedness matrix (GRM) and
rarely referred to as genotype-based numerator relatedness matrix (GNRM).
\textcolor{red}{TODO: consolidate all genotype-based relat* / GRM terminology}
%
Other studies focused on using a variant of GRM in downstream analyses,
particularly in quantitative genetics for
the estimation of heritability,
genome-wide associations, and
genomic prediction
\citep[e.g.][]{nejatijavaremi1997effect, bernardo1999marker, meuwissen2001prediction,
visscher2006assumption, amin2007genomic, vanraden2008efficient,
yu2006unified, astle2009population, yang2010common, zaitlen2013using,
young2018relatedness, feldmann2022average}.
%
All this research has delivered several estimators of GRM.
%
To manage the scale of genome-wide genotype data, these estimators
are now largely based on methods of moments instead of likelihood or
Bayesian methods.
%
In the following we list some of the most common estimators of GRM
and describe their key features.
%
For each we report the numerator relationship matrix
(covariance coefficient) form for diploids.
\textcolor{red}{
TODO: Ensure we don't confuse even more!
      Will list them first and then ensure
      they are all $\bsymb{G}=2\bsymb{K}$ or say what else they are.}
%
While some estimators were developed with a sole focus on relatedness,
others were developed with a focus on downstream applications.
%
For quantitative genetic applications, we should note that
causal loci impacting phenotypes of interest and
marker loci used for data analysis
might not overlap, particularly when using SNP arrays
\citep{deloscampos2015genomic}.
%
SNP arrays also often induce ascertainment bias because marker loci
are selected among more polymorphic and evenly spread loci compared
to all loci.
%
All these points imply that there are several possible GRM, depending
on a source of information, use case, and assumptions.

% SIMPLE ALLELE MATCHING FOR IBS
\paragraph{The simplest estimator of GRM that measures IBS between
individuals is based on allele matching.}
%
IBS allele matching between diploid individuals $i$ and $j$ at locus $l$
is an average of four possible identity comparisions $I$ between the two
alleles of each individual:
%
\begin{equation} \label{eqn:K_m_locus}
  K_{i,j,l}^{m} = \frac{\sum_{o=1}^2 \sum_{o'=1}^2 I\left(w_{i,l,o} = w_{j,l,o'}\right)}{4}.
\end{equation}
%
As an average of indicator variables $I$, $K_{i,j,l}^{m}$ is a
proportion and ranges between $0$ and $1$.
%
Averaged across $n_l$ loci we have:
%
\begin{equation} \label{eqn:K_m}
  K_{i,j}^{m} = \frac{\sum_{l=1}^{n_l} K_{i,j,l}^{m}}{n_l}.
\end{equation}
%
GRM based on IBS allele matching can be efficiently calculated using
matrix operations with an $n_i \times n_l$ genotype matrix $\bsymb{W}$
\citep{vanraden2007efficient, vanraden2008efficient, astle2009population}:
%
\begin{equation} \label{eqn:G_m}
  \bsymb{G}^{m} = 1 + \frac{(\bsymb{W} - \bsymb{1}_{n_i}\bsymb{1}_{n_l}^T)
                            (\bsymb{W} - \bsymb{1}_{n_i}\bsymb{1}_{n_l}^T)^T}{n_l},
\end{equation}
%
where $\bsymb{1}_n$ is $n \times 1$ vector of 1s.
% TODO: Show an example?

% LINEAR MODEL BASED ESTIMATORS OF GRM
\paragraph{Some common estimators of GRM are based on linear
modelling of phenotypic values with genome-wide genotype data.}
%
Development of these approaches was inspired by the seminal work of
\cite{meuwissen2001prediction}, which has shown the informativeness of
genome-wide genotype data for predicting phenotypic values.
%
In the following, we focus on three common estimators of GRM
based on the linear modelling
\citep{vanraden2008efficient, yang2010common, speed2012improved}
and point various extensions before we move to estimators proposed
in different contexts.
%
\cite{vanraden2008efficient} derived estimator of GRM that centres
locus genotypes, calculates their cross-products, and divides the
cross-products by the sum of expected variances of genotypes at loci
(the ratio of sums estimator).
%
\cite{yang2010common} popularised an alternative estimator of GRM
that centres and scales genotypes at every locus before calculating
their cross-products and averaging these across loci
(the average of ratios estimator).
%
\cite{speed2012improved} found a large impact of linkage-disequilibrium
on heritability estimates and proposed an estimator of GRM that
weighs genotypes across correlated loci.

% VANRADEN VERSION (THE RATIO OF SUMS ESTIMATOR)
\paragraph{The ratio of sums estimator of GRM from \cite{vanraden2008efficient} is:}
%
\begin{equation} \label{eqn:G_c}
  \bsymb{G}^{c} = \frac{\bsymb{W}_c\bsymb{W}_c^T}{2\sum_{l=1}^{n_l} p_l q_l},
\end{equation}
%
where $\bsymb{W}_c$ is a centered genotype matrix.
%
A way to interpret this estimator is to take Fisher's phenotype
linear model \eqref{eqn:fisher_pheno_mat} and express the genetic value
of an individual $i$ as a linear combination of its multi-locus genotype
$\bsymb{w}_i$ and allele substitution effects $\bsymb{\alpha}$
\citep{fisher1919correlation, falconer1996introduction}:
%
\begin{equation} \label{eqn:g_walpha}
  g_i^0 = \bsymb{w}_i\bsymb{\alpha}.
\end{equation}
%
We will explain the role of subscript $0$ in the following.
%
Asssuming \textit{iid} normal distribution for allele substitution effects:
%
\begin{equation} \label{eqn:alpha_normal}
  \bsymb{\alpha} \sim \N\left(\bsymb{0}, \bsymb{I}\sigma_{\alpha}^2\right),
\end{equation}
%
and given the observed genotypes $\bsymb{W}$ covariance between the
genetic values $\bsymb{g}^0$ is:
%
\begin{equation} \label{eqn:WWT0}
  \Var\left(\bsymb{g}^0=\bsymb{W}\bsymb{\alpha} | \bsymb{W}\right) = \bsymb{W}\bsymb{W}^T\sigma_{\alpha}^2.
\end{equation}
%
The genotype-based covariance coefficients in $\bsymb{W}\bsymb{W}^T$ have
values from $0$ and above - counting the squared number of alternative
allele matches between individuals.
%
Many individuals are expected to have a positive covariance due to some
number of shared alternative alleles.
%
The diagonal element of $\bsymb{W}\bsymb{W}^T$ for individual $i$ is:
%
\begin{align} \label{eqn:WWT0_ind}
  \bsymb{w}_i\bsymb{w}_i^T & = \sum_{l=1}^{n_l} w_{i,l} w_{i,l} \notag \\
                           & = \sum_{l=1}^{n_l} \left(w_{i,l,1} w_{i,l,1} + w_{i,l,2} w_{i,l,2} + 2w_{i,l,1} w_{i,l,2}\right) \notag \\ 
                           & = \bsymb{w}_{i,1}\bsymb{w}_{i,1}^T + \bsymb{w}_{i,2}\bsymb{w}_{i,2}^T + 2\bsymb{w}_{i,1}\bsymb{w}_{i,2}^T, 
\end{align}
%
showing the connection with the pedigree-based variance coefficient $1 + F_i$
in \eqref{eqn:g_ind_var}.
%
The off-diagonal element of $\bsymb{W}\bsymb{W}^T$ for individuals
$i$ and $j$ is similarly:
%
\begin{align} \label{eqn:WWT0_pair}
  \bsymb{w}_i\bsymb{w}_j^T & = \sum_{l=1}^{n_l} w_{i,l} w_{j,l} \notag \\
                           & = \bsymb{w}_{i,1}\bsymb{w}_{j,1}^T + \bsymb{w}_{i,1}\bsymb{w}_{j,2}^T + \bsymb{w}_{i,2}\bsymb{w}_{j,1}^T + \bsymb{w}_{i,2}\bsymb{w}_{j,2}^T,
\end{align}
%
showing the connection with the pedigree-based covariance coefficient
$2K_{i,j}$ in \eqref{eqn:g_pair_cov} and pedigree-based kinship coefficient
$K_{i,j}$ in \eqref{eqn:kinship_pair}.
%
Hence, the $\bsymb{w}_{i,o}\bsymb{w}_{j,o'}^T$ and $\bsymb{w}_i\bsymb{w}_j^T$
are respectively genome and individual relatedness coefficients based on
observed genotypes according to the assumed model in \eqref{eqn:g_walpha}.
% TODO: Show an example?
%
Compared to pedigree data, observing genotype data informs how much the
value of individuals' genomes deviate from the expectation given the number
of alternative alleles and how much they correlate given the shared number
of alternative alleles.
%
Observing genotype data also reveals parent average and Mendelian sampling
components of the genotypes \eqref{eqn:wright_geno}:
$\bsymb{w}_i = \nicefrac{1}{2}\bsymb{w}_{m(i)} + \nicefrac{1}{2}\bsymb{w}_{f(i)} + \bsymb{w}_i^r$,
where $\bsymb{w}_i^r = \bsymb{w}_i - \nicefrac{1}{2}\left(\bsymb{w}_{m(i)} + \bsymb{w}_{f(i)}\right)$
is the Mendelian sampling component.
%
% This genetic relatedness is trait agnostic.
%
% Trait specific genetic relatedness is estimated by fitting the
% model such as \eqref{eqn:fisher_pheno_mat} to the data and
% some post-processing \citep{gianola2020inferring}.
% TODO: gianola2020inferring infers p(G(\beta) | y) as trait-specific
%       genetic relatedness (in genomic era) and made me wonder how 
%       p(G(\beta) | y) and Var(g | y) are related!

Because we encoded genotypes as $0$, $1$, or $2$, the genetic values
$\bsymb{g}^0$ in \eqref{eqn:g_walpha} are expressed as a deviation from
the reference homozygote individual ($w_{i,l}=0$ for all loci).
%
It is common to centre covariates in linear models to estimate the intercept
as the overall mean of population or some other reference level at the
``centre'' of sampled data.
%
When such centering is applied to the genotype matrix $\bsymb{W}$, we model
genetic values as a deviation from the population ``centre'' as initially
defined by \cite{fisher1919correlation}.
%
Subtracting and adding $\overline{\bsymb{W}}\bsymb{\alpha}$ from
$\bsymb{g}^0$ gives:
%
\begin{align} \label{eqn:fisher_pheno_mat_walpha}
  \bsymb{y} & = \bsymb{Xb}^0 + \bsymb{Z}(\bsymb{W}\bsymb{\alpha}
                             - \overline{\bsymb{W}}\bsymb{\alpha}
                             + \overline{\bsymb{W}}\bsymb{\alpha})
                             + \bsymb{e} \notag \\ 
            & = \bsymb{Xb} + \bsymb{Z}\bsymb{W}_c\bsymb{\alpha} + \bsymb{e} \notag \\
            & = \bsymb{Xb} + \bsymb{Z}\bsymb{g} + \bsymb{e},
\end{align}
%
where
$\bsymb{Xb} = \bsymb{Xb}^0 + \bsymb{Z}\overline{\bsymb{W}}\bsymb{\alpha}$,
$\bsymb{W}_c = \bsymb{W} - \overline{\bsymb{W}}$,
and variance is:
%
\begin{equation} \label{eqn:WcWcT}
  \Var\left(\bsymb{g}=\bsymb{W}_c\bsymb{\alpha} | \bsymb{W}_c\right) = \bsymb{W}_c\bsymb{W}_c^T\sigma_{\alpha}^2.
\end{equation}
%
This centering shifts the reference point for genotype-based relatedness, 
which can now be negative, zero, or positive.
%
Different reference points can be used.
%
\cite{vanraden2008efficient} suggested use of twice the allele frequency
from pedigree founders because he was comparing genotype-based $\bsymb{G}$
with the pedigree-based $\bsymb{G}$, but also suggested use of twice the
sample allele frequency \eqref{eqn:geno_exp}.
%
With the centering matrix $\bsymb{C}=\bsymb{I}_{n_i} - \frac{1}{n_i}\bsymb{1}_{n_i}\bsymb{1}_{n_i}^T$,
we sample-centre the genotypes as $\bsymb{C}\bsymb{W}$ giving:
%
\begin{equation} \label{eqn:CWWTCT}
  \Var\left(\bsymb{g}=\bsymb{C}\bsymb{W}\bsymb{\alpha} | \bsymb{W}\right) = \bsymb{C}\bsymb{W}\bsymb{W}^T\bsymb{C}^T\sigma_{\alpha}^2.
\end{equation}
%
When $n_l > n_i$, the rank of $\bsymb{C}\bsymb{W}$ is at most $n_i - 1$.
%
Note that estimating allele frequencies from the analysed data can bias
estimates of relatedness \citep{ritland1996estimators, astle2009population, toro2011note, ackerman2017estimating, weir2017unified},
though \eqref{eqn:CWWTCT} is the genetic covariance matrix based on the
model \eqref{eqn:fisher_pheno_mat_walpha}.
%
Centering removes the impact of non-polymorphic loci from
\eqref{eqn:CWWTCT} because the effect of these loci is captured by the
model intercept in \eqref{eqn:fisher_pheno_mat_walpha}.
%
For loci with low frequency of alternative alleles, heterozygotes and alternative
homozygotes have a greater impact on relatedness than for loci with intermediate
frequency.
%
In other words, when an alternative allele is rare, sharing such alleles indicates
greater relatedness.
% TODO: Check these two last sentences!
%       I am a bit confused if this is indeed what is going on - isn't this impact
%       only happening when we scale centred genotypes by SD? For sure centering
%       doesn't change allele prior.
%
The centering hence expresses relatedness between individuals relative to the
average relatedness in a chosen ``reference'' (sub-)population and is as such
connected to the within, between, and across population relatedness concepts
\eqref{eqn:fst} as well as IBS and IBD \eqref{eqn:fibd_fibs}
\citep{wright1949genetical, jacquard1975inbreeding, powell2010reconciling,
legarra2015ancestral, legarra2016comparing}.
%
However, these connections are not well defined and rely on arbitrarily
chosen ``reference'' (sub-)population and corresponding allele frequencies
\citep{legarra2016comparing, powell2010reconciling, speed2015relatedness}.
%
Because centering reduces the rank of $\bsymb{G}^{c}$, practitioners usually
add a small value to the diagonal of $\bsymb{G}^{c}$ to enable fitting the model 
\eqref{eqn:fisher_pheno_mat_walpha} \citep{vanraden2008efficient},
though an equivalent model with marker effects $\bsymb{\alpha}$ can be fitted
instead of \eqref{eqn:fisher_pheno_mat_walpha},
and genetic values $\bsymb{g}$ can be then estimated from genotypes
and marker effect estimates \citep{stranden2009derivation}.
%
\cite{stranden2011allele} study in detail different centering approaches
and show that while centering changes the GRM, estimates of genetic values
are independent of centering when intercept is included in the model.
%
% TODO: zhang2023biobank (sup 3) show scale, data shift, and constant shift invariance
%       that clarify the above points even further!

Finally, to obtain a genotype-based NRM that is similar to the pedigree-based NRM,
\cite{vanraden2008efficient} scaled $\bsymb{W}_c\bsymb{W}_c^T$ with
the sum of expected variances of locus genotypes assuming Hardy-Weinberg
equilibrium, $2\sum_{l=1}^{n_l} p_l q_l$ \eqref{eqn:geno_var}, giving
\eqref{eqn:G_c}.
% TODO: Show an example?
%
This scaling changes the variance parameter $\sigma_{\alpha}^2$
in \eqref{eqn:CWWTCT} to $2\sum_{l=1}^{n_l} p_l q_l\sigma_{\alpha}^2$,
referred as ``genomic variance'' - an attemp to estimate the genetic variance
$\sigma_g^2$ in \eqref{eqn:g_normal}.
%
This approach however ignores the miss-match between causal and
marker loci \citep{deloscampos2015genomic} and the contribution of
inbreeding and linkage-disequilibrium to $\sigma_g^2$ \citep{lara2022temporal};
see also \cite{shi2016contrasting, hou2019accurate, rawlik2020snp}.
%
To see the impact of ignoring inbreeding, assume a population of
fully inbred individuals (self-pollinating plants such as wheat),
hence $F_l = 1$ for all loci and ``genomic variance'' is
$4\sum_{l=1}^{n_l} p_l q_l\sigma_{\alpha}^2$, while scaling with
$2\sum_{l=1}^{n_l} p_l q_l$ will lead to an estimate of ``genomic
variance'' in an outbred population instead of inbred population.
%
How well a linear model with $\bsymb{G}^{c}$ fits the observed
phenotypic values is partly a function of the number of markers.
%
\cite{endelman2012shrinkage} proposed a shrinkage estimator of
$\bsymb{G}^{c}$ when a limited number of markers is used.

% AMIN/YANG/GCTA VERSION (THE AVERAGE OF RATIOS)
\paragraph{The average of ratios estimator of GRM popularised by
\cite{yang2010common} is:}
%
\begin{equation} \label{eqn:G_s}
  \bsymb{G}^{s} = \frac{\bsymb{W}_s \bsymb{W}_s^T}{n_l},
\end{equation}
%
where $w_{s,i,l} = \left(w_{i,l} - 2 p_l\right) / \sqrt{2 p_l q_l}$
is standardised vector of genotypes at locus $l$ assuming Hardy-Weinberg
equilibrium (\ref{eqn:geno_prob_hwe}, \ref{eqn:geno_exp}, \ref{eqn:geno_var}).
%
$\bsymb{G}^{s}$ is an individual-level estimator variant of the
Li and Horvitz population-level inbreeding coefficient estimator
\eqref{eqn:LiHorvitz_f}
\citep{li1953some, ritland1996estimators, astle2009population}.
%
Since $w_{s,i,l}$ are standardised genotypes, $\bsymb{G}^{s}$ is based on
correlations between genotypes of individuals following \eqref{eqn:geno_cov}
and averaged across loci \cite{astle2009population, speed2015relatedness}.
%
To derive this correlation, rearrange \eqref{eqn:fibd_fibs_genotype} to:
%
% Pr(aa) = F_IBD,i Pr(a) + (1-F_IBD,i) Pr(a)^2
%        = F_IBD,i Pr(a) + Pr(a)^2 - F_IBD,i Pr(a)^2
%        = F_IBD,i (Pr(a) - Pr(a)^2) + Pr(a)^2
% F_IBD,i = (Pr(aa) - Pr(a)^2) / (Pr(a) - Pr(a)^2)
%         = (Pr(aa) - Pr(a)^2) / (Pr(a) (1 - Pr(a)))
%
\begin{align} \label{eqn:fibd_as_cor_probs}
  F_{IBD,i,l} & = \frac{\Pr\left(w_{i,l} = 2\right) - \Pr\left(w_{i,l,o} = 1\right)^2}
                       {\Pr\left(w_{i,l,o} = 1\right) \left(1 - \Pr\left(w_{i,l,o} = 1\right)\right)},
\end{align}
%
and noting that
$\Pr\left(w_{i,l} = 2\right) = \E\left(w_{i,l,o},w_{i,l,o'}\right)$,
$\Pr\left(w_{i,l,o} = 1\right) = \E\left(w_{i,l,o}\right) = p_l$ \eqref{eqn:allele_freq},
$\Pr\left(w_{i,l,o} = 1\right) (1 - \Pr\left(w_{i,l,o} = 1\right))
  = \Var\left(w_{i,l,o}\right) = p_l q_l$ \eqref{eqn:geno_var},
and
$\Pr\left(w_{i,l} = 2\right) - \Pr\left(w_{i,l,o} = 1\right)^2 
  = \E\left(w_{i,l,o},w_{i,l,o'}\right) - \E\left(w_{i,l,o}\right)\E\left(w_{i,l,o}\right)$  \eqref{eqn:allele_cov}
%  = \E\left(\left(w_{i,l,o}  - \E\left(w_{i,l,o}\right)\right)
%            \left(w_{i,l,o'} - \E\left(w_{i,l,o'}\right)\right)\right)$
%
we have:
%
\begin{align} \label{eqn:fibd_as_cor}
  F_{IBD,i,l} & = \frac{\Cov\left(w_{i,l,o}, w_{i,l,o'}\right)}
                       {\sqrt{\Var\left(w_{i,l,o}\right) \Var\left(w_{i,l,o'}\right)}}\\
              & = \Cor\left(w_{i,l,o}, w_{i,l,o'}\right), \notag \\
              & = w_{s,i,l,o} w_{s,i,l,o'}. \notag
\end{align}
%
Expanding \eqref{eqn:fibd_as_cor} to genotype of an individual or
genotypes for a pair of individuals and averaged across loci gives
$\bsymb{G}^{s}$.
%
An earlier reference to this method of moments estimator is \cite{amin2007genomic},
who cite \cite{leutenegger2003estimation}, though
\cite{leutenegger2003estimation} used maximum likelihood with a Hidden Markov
Model (HMM) to estimate inbreeding coefficients from observed genotype data
along chromosomes.
%
\cite{amin2007genomic} aimed to estimate kinship coefficients, but they centred
genotypes with $p_l$ instead of $2p_l$ \eqref{eqn:geno_exp} and scaled by
$\sqrt{p_l q_l}$ instead of $\sqrt{4 p_l q_l}$ \eqref{eqn:geno_var}; $4$ to
estimate kinship coefficients instead of twice the kinship coefficients
\citep{astle2009population}.
%
As with the $\bsymb{G}^{c}$ estimator \eqref{eqn:G_c}, centering in
\eqref{eqn:G_s} gives negative, zero, or positive relatedness coefficients;
and reduces the matrix rank.
%
This estimator is unstable for very low frequencies and undefined for
fixed loci.
% TODO: Is the word "unstable" used here correctly? I mean the estimators
%       is defined the way it is defined, but with very low frequency we
%       get extemely large values when we have
%       (w[i,j]-tiny[j]) / sqrt(2*tiny[j]*(1-tiny[j])
%
While the $\bsymb{G}^{c}$ implies \textit{iid} normal distribution for
allele substitution effects \eqref{eqn:alpha_normal}, $\bsymb{G}^{s}$ 
implies heterogeneous variance \citep{speed2012improved}:
%
\begin{equation} \label{eqn:alpha_normal_hetero}
  \alpha_l \sim \N\left(0, \sigma_{\alpha}^2 / \left(2 p_l q_l\right) \right).
\end{equation}
%
Scaling therefore implies that allele substitution effects could
deviate more for loci with rare alleles than for loci with common alleles.
% TODO: Show an example?
%
\citep{bouwman2017estimated} evaluates impact of
this prior \eqref{eqn:alpha_normal_hetero} and
the \textit{iid} prior \eqref{eqn:alpha_normal} on
the estimates of allele substitution effects and genetic values.
%
Further research on estimating genome-wide allele substitution effects
with other functional relationships with the frequency of alleles has been
initiated by \cite{speed2012improved} and with selection by
\cite{zeng2018signatures} and \cite{speed2022snp}.
%
Lastly, \cite{yang2010common} proposed a correction of
\eqref{eqn:G_s} due to miss-match between causal and marker loci -
driven by sampling error and differences in linkage-disequilibrium
between these two sets of loci (see also \cite{speed2012improved}
and \cite{deloscampos2015genomic}).

% SPEED/LDAK VERSION (THE AVERAGE OF WEIGHTED RATIOS)
\paragraph{Genome-wide markers densely tag genome variation
enabling the capture of unknown causal loci, yet not all genome
regions are equivalently tagged.}
%
\cite{speed2012improved} and \cite{speed2017reevaluation} found
that uneven tagging with respect to linkage-disequilibrium can
impact estimates of heritability
$\left(h^2=\nicefrac{\sigma_g^2}{\left(\sigma_g^2 + \sigma_e^2\right)}\right)$.
%
They proposed weighing markers according to local linkage-disequilibrium,
changing the GRM estimator \eqref{eqn:G_s} to:
%
\begin{equation} \label{eqn:G_u}
  \bsymb{G}^{u} = \frac{\bsymb{W}_u \bsymb{W}_u^T}{n_l},
\end{equation}
%
where $w_{u,i,l} = w_{s,i,l} \sqrt{u^*_l}$ and
$u^*_l = u_l \frac{n_l}{\sum_{j=1}^{n_l} u_j} > 0$.
%
The notation $u$ indicates that \eqref{eqn:G_u} attempts to manage
correlations between genotypes at marker loci.
%
The $u_l$ weighs the locus $l$ so that loci in a region with
linkage-disequilibrium have equalised prior contribution to regional
genetic values, implying this prior for allele substitution effects:
%
\begin{equation} \label{eqn:alpha_normal_hetero_ld}
  \alpha_l \sim \N\left(0, \sigma_{\alpha}^2 u^*_l / \left(2 p_l q_l\right) \right).
\end{equation}
%
% TODO: Do we need to redefine g with a QTL and two markers in correlation with it
%       and then follow Boichard's trick to show how marker effects relate to the QTL
%       (show this by setting true moedl for y = g + eps = qbeta + eps, then analysis
%       model y = mu + w1alpha + w2alpha + e, setup MME, invert 3x3 LHS symbolically,
%       express one marker solution as a function of inv(LHS)_row times RHS (where
%       we express y in RHS as qbeta + eps) and then we can do w2=w1*r12 and show that
%       by introducting weights u1 and u2 we can change the contribution to WHAT?
% ALRIGHT I AM GETTING INTO A RABBIT HOLE HERE! It could help understanding later
% when we deal with LD-prunned PCA though ... LEAVE IT FOR LATER!
%
% To see this, imagine a region with two linked markers and genetic values
% explained by these markers 
% $\bsymb{g}^r = \bsymb{w}^1\beta_1 + \bsymb{w}^2\beta_2$ with
% $\bsymb{w}^l = \bsymb{w}_{s,:,l}$ and
% $\Var\left(\bsymb{g}^r | \bsymb{w}^1, \bsymb{w}^2\right) =
%   (\bsymb{w}^1\bsymb{w}^{1^T} +
%    \bsymb{w}^2\bsymb{w}^{2^T} +
%    2\bsymb{w}^1\bsymb{w}^{2^T})\sigma_{\beta}^2$,
% where $\bsymb{w}^1\bsymb{w}^{2^T}$ is correlation coefficient between
% genotypes at the two loci.

% GRM "extensions"
\paragraph*{There is large number of extensions of the above three
GRM estimators for quantitative genetic analyses in different settings.}
%
Some of the extensions related to  additive genetic modelling include:
%
mutations between generations \citep{casellas2013accounting},
%
X-chromosome or haplo-diploids \citep{druet2020theoretical},
%
mitochondrial DNA \citep{mafrafortuna2023accounting},
%
% dominance \citep{there is a lot of work here} % skipping to focus just on additive genetic model(s)
%
imprinting \citep{nishio2015genomic},
%
% epistasis \citep{there is a lot of work here} % skipping to focus just on additive genetic model(s)
%
pools of individuals from different families using family allele
frequencies instead of individual's genotypes
\citep{johnston2013fish, ashraf2014association, ashraf2016estimating},
%
polyploids \citep{slater2016improving, bilton2024construction},
%
low-depth sequencing data \citep{dodds2015construction}, and
%
combining pedigree and genotype data 
\citep{legarra2009a, christensen2010genomic, legarra2014single}.
%
With the latter, care is needed to account for the differences in pedigree
and genomic sources of genetic information, particularly regarding assumed
background relatedness of pedigree founders,
pedigree incompletness, and
centering of genotypes
\citep{legarra2015ancestral, legarra2016comparing, masuda2022unknown}.
%
Several of these extensions are implemented in the R package AGHmatrix
\citep{amadeu2023aghmatrix}.

% TODO: Mention somewhere to weight the GRM by the effects of markers
%       (VanRaden suggested this already, but I feel someone musth have
%        suggested that before him?!)
% Weighted single-step genomic best linear unbiased prediction integrating variants selected from sequencing data by association and bioinformatics analyses
% https://link.springer.com/article/10.1186/s12711-020-00568-0
%
% Speed must have had something in this space too?
%
% --> this is all going towards trait-specific genetic relatedness,
%     and this could be mentioned where I tried citing gianola2020inferring

% TODO: Next - parse Kenneth's thesis and summarise his summary;)
% Genomic genetic groups
% * TODO: There is quite a bit of work (masuda2022unknown brough things together), but NTNU stuff
%         from Kenneth (see below) goes beyond that!
% * TODO: Genetic Group Animal Models in the Genomics Era
%         https://ntnuopen.ntnu.no/ntnu-xmlui/handle/11250/2778383

% TODO: Cite Cantet's ancestral regression model somewhere here or later?
% Since inheritance of parental genomes effectively samples grandparental genomes
% Cantet et al. looked at expanding the standard two-generation pedigree model
% for genetic values of two parents and a progeny to three-generation pedigree model
% of four grandparents, two parents, and a progeny - to explain/model Mendelian sampling
% variation.

% * TODO: Goudet has heiferstat R package
% * TODO: Does Ochoa have an R package too?

% Exploiting rank-deficiency with low Ne
% With growing size of genomic data, the GRM based on SNP array genotypes is often rank deficent
% * TODO: Inexpensive Computation of the Inverse of the Genomic Relationship Matrix in Populations with Small Effective Population Size
%         https://pubmed.ncbi.nlm.nih.gov/26584903/
% * TODO: On the equivalence between marker effect models and breeding value models and direct genomic values with the Algorithm for Proven and Young
%         https://gsejournal.biomedcentral.com/articles/10.1186/s12711-022-00741-7
% * TODO: Large-scale genomic prediction using singular value decomposition of the genotype matrix --> SVD
%         https://gsejournal.biomedcentral.com/articles/10.1186/s12711-018-0373-2
% * TODO: GT-BLUP from LUKE folks

% TODO
\paragraph{Fitting the model \eqref{eqn:fisher_pheno_mat_walpha} with a
GRM has reinvigorated quantitative genetics.}
%
It enables detailed study of fundamental topics such as
genetic variance and heritability
\citep{falconer1996introduction, visscher2008heritability}.
%
Furthemore, it provides higher accuracy of estimated genetic values
than when PRM is used
\citep{nejatijavaremi1997effect, meuwissen2001prediction, vanraden2008efficient}.
%
It also makes such analyses possible when pedigrees are not available
\citep[e.g.][]{bernardo1999marker, johnston2022taking}.
%
While the use of GRM enables all these analyses with accuracy, the active
search for the correct GRM has shown that different GRMs might be optimal
for different types of analyses or settings
\citep[e.g.][]{speed2015relatedness, legarra2016comparing, weir2018how, feldmann2022average}.
%
TODO: ochoa's work on model miss-specification (LMM still does a good job!)
\cite{ochoa2021estimating}
%
With respect to the estimation of heritability
% TODO: point that genomic variance is not genetic variance so searching
%       for optimal GRM based on that is not the right path ...
\citep[e.g][]{shi2016contrasting, hou2019accurate, rawlik2020snp, lara2022temporal}.
%
% TODO: Mention huge body of work on PCA corrections for GRM, env confounding \citep{young2018relatedness}, etc.

% * TODO: \cite{weir2017unified} A Unified Characterization of Population Structure and Relatedness
%
% * TODO: \cite{weir2018how} How to estimate kinship
%
% * TODO: It seems that the work of Weir and of Ochoa is the most complete - study them closely, including all the slides from Weir!

TODO: Check R packages:
  * Relatedness (Laporte and Mary-Huard 2017, Laporte et al. 2017)
  * R package Jacquard which implements estimation of Jacquard’s coefficients and derived quantities by constrained least squares

% All these ignore linkage or LD though
\paragraph*{TODO.}

\cite{leutenegger2003estimation} used HMM to find IBD segments based on genome-wide data and then estimate inbreeding

% Modern IBD
% * TODO: \cite{powell2010reconciling} Reconciling the analysis of IBD and IBS in complex trait studies 
% * TODO: \cite{visscher2006assumption} Assumption-Free Estimation of Heritability from Genome-Wide Identity-by-Descent Sharing between Full Siblings
%         He has a nice summary paragraph on Variance of genome-wide IBD sharing in M&M
% * TODO: hickey2013genomic Genomic evaluations using similarity between haplotypes
% * TODO: zaitlen2013using Using Extended Genealogy to Estimate Components of Heritability for 23 Quantitative and Dichotomous Traits
% * TODO: luan2014genomic Genomic prediction based on runs of homozygosity
% * TODO: \cite{thompson2013identity} Identity by descent: variation in meiosis, across genomes, and in populations 
% * TODO: \cite{browning2012identity} Identity by Descent Between Distant Relatives: Detection and Applications},
% * TODO: \cite{hill2011variation} Variation in actual relationship as a consequence of Mendelian sampling and linkage 
% * TODO: \cite{garciacortes2013variance}
% * TODO: \cite{edwards2015two} Two molecular measures of relatedness based on haplotype sharing
% * TODO: \cite{pook2019haploblocker} {HaploBlocker}: Creation of Subgroup-Specific Haplotype Blocks and Libraries
% * TODO: \cite{white2020effect} Effect of heterogeneity in recombination rate on variation in realised relationship 
% * TODO: \cite{saada2020identity} Identity-by-descent detection across 487,409 British samples reveals fine scale population structure and ultra-rare variant associations
% * TODO: There was a paper on searching for IBD segments quickly (Gusev GERMLINE, but also more recent for BioBank scale --> huge domain, including Peer, Sharmi, ...)
% * TODO: tsambos2022efficient Eﬃcient analysis of genetic ancestry in population-sized datasets
% * TODO: Browning & Browning (2023) Biobank-scale inference of multi-individual identity by descent and gene conversion
% * TODO: Weave in Huang et al. (2024) ARG&IBD https://www.biorxiv.org/content/10.1101/2024.03.07.583855v1
% * TODO: Kang et al. (2020) Performance of Genomic Prediction using Different Multi-Allelic Genomic Relationship Matrices for Genotyping-by-Sequencing (GBS) Data
%        https://www.researchgate.net/publication/355862804_Performance_of_Genomic_Prediction_using_Different_Multi-Allelic_Genomic_Relationship_Matrices_for_Genotyping-by-Sequencing_GBS_Data
%        He has lots on short multi-allelic haps, but there is more previous work on this
%        https://www.researchgate.net/profile/Jie-Kang-6/publication/355861564_Construction_of_Multi-Allelic_Genomic_Relationship_Matrices_using_Genotyping-by-Sequencing_GBS_Data_Overview_Background/links/6181d7160be8ec17a963874d/Construction-of-Multi-Allelic-Genomic-Relationship-Matrices-using-Genotyping-by-Sequencing-GBS-Data-Overview-Background.pdf
%        https://www.researchgate.net/publication/355867845_Can_short_haplotypes_improve_genomic_prediction

TODO: cite paper from Alex Young on what WGS data gives us comapred to SNP arrays
      (SNP arrays have quite polymorphic loci meaning that they largely represent
       older branches, while WGS data gives us also information about recent branches)
       young2022discovering Discovering missing heritability in whole-genome sequencing data

TODO: cite also Yiang/Yengo/Visscher WGS paper(s)? Probably don't need to - that MAF
      and LD weighting business is not what we want (see Shi et al, Hou et al., and Lara et al.),
      but an important point with WGS data is that we are not getting much more
      accuracy of genomic predictions (at least not in breeding settings), which
      could be down to inappropriate modelling of the data - for example we are
      not "accounting/modelling" decreasing Ne in breeding populations, which means
      that ARG is very "shallow" with lots of recent MRCAs etc.

% --- PHYLOGENETIC TREE -------------------------------------------------------

\subsection{Phylogeny-based relatedness}

\paragraph{Phylogenetic tree}
\begin{itemize}
  \item \textcolor{red}{SAY WHAT A PHYLOGENY IS!?}
  \item Phylogenetic relatedness matrix was initiated by \cite{lynch1991methods}
  \item Efficient algorithm for covariance and inverse: General quantitative genetic methods for comparative biology: phylogenies, taxonomies and multi-trait models for continuous and categorical characters \citep{hadfield2010general}
  \item \citep{hadfield2010general} says: "In a phylogenetic context the equivalent matrix is equal in dimension to the number of species at the tips of the phylogeny. In this case the elements Aij are equal to the length of the path from the most recent common ancestor of species i and j to the root of the phylogeny. Generally, the length of the path from the tips to the root of the phylogeny is scaled to unity so that the matrix is the correlation matrix with all the diagonal elements being 1."
  \item there is also Slatkin's paper on inbreeding and coalescence times \cite{slatkin1991inbreeding}
  \item The above is for species trees, but we have work for haplotype trees in: Hierarchical Modelling of Haplotype Effects on a Phylogeny \cite{selle2021hierarchical}
\end{itemize}

\cite{slatkin1991inbreeding} showed that $F_{ST} = (\E(T_B) - \E(T_W)) / \E(T_B)$
where $\E(T_B)$ is the expected coalescence time of two lineages sampled in two
different populations and $\E(T_W)$ for lineages sampled from the same population.

Felsenstein 1973, maximum-likelihood estimation of evolutionary trees from continuous characters
Felsenstein 1981 evolutionary trees from gene frequencies and quantitative characters

% TODO: Wuif & Hein (1997) On the Number of Ancestors to a DNA Sequence (maybe use later or ...)
%       https://academic.oup.com/genetics/article/147/3/1459/6054122

TODO: there is lots more in this community! Ann etc.
% TODO: Ho & Ane (2014)
%   A Linear-Time Algorithm for Gaussian and Non-Gaussian Trait Evolution Models 
%   https://academic.oup.com/sysbio/article/63/3/397/1649891

TODO: some of this stuff below will have to go to the coalescent/gene tree sub-section

% TODO: Mention that phylogeny- and coalescent-based genetic relatedness is effectively
%   the same thing - do we need a separate coalescent-based sub-section capturing
%   Malecot, Slatkin, and others (who?)? (doh)?
%
% TODO: Prior work by clustering haplos and estimating their effects - see \citep{selle2021hierarchical} et al. & Zollner citations
%
% TODO: 
%   Zollner & Pritchard (2005)
%   Coalescent-Based Association Mapping and Fine Mapping of Complex Trait Loci
%   https://academic.oup.com/genetics/article/169/2/1071/6060255?login=true
%
% TODO: Edge & Coop (2019)
%   Reconstructing the History of Polygenic Scores Using Coalescent Trees
%   https://academic.oup.com/genetics/article/211/1/235/5931145#
%
% TODO: \citep{selle2021hierarchical}
%   Hierarchical Modelling of Haplotype Effects on a Phylogeny
%   https://pubmed.ncbi.nlm.nih.gov/33519886

% COPIED FROM GENO SUB_SECTION: There is another popular set of F-statistics \citep{patterson2012ancient},
% which are defined as covariance \citep{peter2016admixture} and have a close
% relationship with the Principal Component Analysis (PCA) of allele and genotype
% variation \citep{peter2022geometric}.
% --> connect peter2016admixture with Slatkin
% --> connect peter2022geometric with mcvean2009genealogical

% TODO: Weave in the F-stat for general hierarchy/tree part
%       The relationship between these coefficients is
%       $F_{IT} = F_{IS} + (1 - F_{IS}) F_{ST}$ or
%       $1 - F_{IT} = \left(1 - F_{IS}\right)\left(1 - F_{ST}\right)$.
%
%       This relationship can be expanded to
%       an arbitrary number of levels \citep{ochoa2021estimating} and
%       hierarchical (tree-like population structure \citep{maryhuard2023fast}.

% TODO: \cite{schraiber2024unifying} drew connections between pedigree,
%       genomic, phylo, and ARG/tree NRMs
%       https://www.biorxiv.org/content/10.1101/2024.02.10.579721v1.full.pdf
%       https://read.readwise.io/read/01hpghf1ejvxtwzw26b14bk0zd

% TODO: Paabo (2003) The mosaic that is our genome
%       https://doi.org/10.1038/nature01400
%
% TODO: Ding et al. (2023) Polygenic scoring accuracy varies across the genetic ancestry continuum
%       https://www.nature.com/articles/s41586-023-06079-4
%       --> here is a nice showcase why we want to manage genetic distances between individuals well

TODO: Mention concept of ancestral allele and mutation and link to reference and
      alternative allele

% --- COALESCENT - GENE TREE --------------------------------------------------

\subsection{Coalescent-based relatedness}

\paragraph{Coalescent TODO}

\cite{kingman1982on}
\cite{kingman1982the}
\cite{hein2004gene}

% --- ARG ---------------------------------------------------------------------

\subsection{ARG-based relatedness}

\paragraph{ARG TODO}

coalescent with recombination

Say that this builds on Malecot and coalescent

Now we can actually start to do them at scale (already mentioned in intro) - still make a good connection


TODO: kelleher2016efficient, kelleher2018efficient

TODO: Cite tagami2024tstrait somewhere

TODO: use \cite{speidel2019method}

TODO: use \cite{kelleher2019inferring}

TODO: use \cite{zhang2023biobank}

TODO: use \cite{harris2023using}

TODO: use \cite{deng2024robust}

What is ARG

TODO: cite \citep{wong2023general}

TODO: cite \citep{lewanski2024era}

TODO: cite \citep{brandt2024promise}

Technically, they obtain a RM for each tree by computing a weighted average of the RM based on branches by their length and extend it to the complete ARG by weighting each tree-based RM by its total branch length times the number of base pairs covered.
%
\citet{zhang2023biobank} computed their BRM as the expectation of the SRM that would be obtained using sequencing data, assuming that mutations are sampled uniformly over the area of the ARG via Monte Carlo sampling of mutations.
%


many of these papers use the term genealogy, genetic genealogy, genome-wide genealogy,
which is unfortunate because pedigree and genealogy are synonyms, adding to terminological confusion

TODO: Allen \& McAvoy (2024) coalescent in finite pops ... https://arxiv.org/pdf/2207.02880.pdf

% TODO: nowbandegani2023extremely Extremely sparse models of linkage disequilibrium in ancestrally diverse association studies

---

From \cite{slatkin2002modern} describing work of Malecot
The coalescent is not simply a phylogenetic scheme. It is an abstraction based on the
properties of a set of lineages when followed backwards in time in a finite population.
Here again. Malecot aniticipated the key idea. Figure 1. 1(b) shows one of his favorite
drawings. It represents the ancestry of the two genes of an individual back to a single
common ancestor. This Ilgure draws its inspiration from Wright's path coefficients.
However, Malecot (1941) substituted his coefficient of consanguinity to Wright's
coefficient of inbreeding. thus replacing identity by state (based on shared allelic states)
by identity by descent (based on shared ancestry). Then, and only then, did he introduce
allelic differences between genes by allowing a potentially infinite number of mutations
to occur along these lines of descent. This was the start of a new approach to population
genetics that was later developed by Kimura, Crow, Ewens, Watterson, Kingman, and many others.

---

\textcolor{red}{SAY WHAT AN ARG IS!? - cite Yan's pre-print}

Ancestral recombination graph (ARG) represents coalescent, recombination, and mutation events for a set of haplotypes (Hudson, 1991). The information contained in ARG can be represented in two ways. The traditional approach is using a 
sequence of local/gene trees for every locus along the haplotypes (Griffiths and Marjoram, 1997). These trees descrie how the haplotypes differ due to recombination and mutation events. Recombinations change the topology of the local trees. Mutations along the branches of the trees give rise to the variation between

A novel approach is to instead focus on ???

These edges are shorter (longer) between older (recent) nodes because recombination had more (less) opportunity to split them.

The above relatedness metrics all measure similarity between individuals' genomes.

ARG is an ultimate objects for this and describing variation/similarity between individuals and populations at a broad-scale and fine-scale level is increasingly important

Recent developments in encoding ARGs, i.e. tree sequences, open the door to more efficient calculations of genetic relatedness. Brief intro to tree sequences

Tree sequences are an efficient encoding of the \gls{arg}.
In brief, a tree sequence described a series of correlated ``local trees'', with each consecutive tree describing consecutive portions of the genome.
Each node in the tree thus represents a portion of an individual's chromosome, with edges in the tree describing the genealogical relationships between individuals in the sample population.
Each tree can be augmented with information on genetic variation by describing where mutations occur on the branches of the tree.
Importantly, adjacent trees are highly correlated in that they share a similar structure so it is possible to encode the tree sequence efficiently by just storing the differences in structure between consecutive trees.
The tree sequence data structure provides not just efficient storage of genetic variation data but also a mechanism by which to compute summary quantities based on the \gls{arg}, such as the \gls{grm}.

% TODO; \cite{wakeley2016coalescent} Coalescent and models of identity by descent (I have a highighted copy in Reader)
% TODO: \cite{thompson2013identity}
% TODO: Where should we \cite{donnelly1983probability}?
%       The probability that related individuals share some section of genome identical by descent
% Gene identity by descent (IBD) is a fundamental concept that underlies genetically mediated similarities among relatives. Gene IBD is traced through ancestral meioses and is defined relative to founders of a pedigree, or to some time point or mutational origin in the coalescent of a set of extant genes in a population. The random process underlying changes in the patterns of IBD across the genome is recombination, so the natural context for defining IBD is the ancestral recombination graph (ARG), which specifies the complete ancestry of a collection of chromosomes. The ARG determines both the sequence of coalescent ancestries across the chromosome and the extant segments of DNA descending unbroken by recombination from their most recent common ancestor (MRCA). DNA segments IBD from a recent common ancestor have high probability of being of the same allelic type. Non-IBD DNA is modeled as of independent allelic type, but the population frame of reference for defining allelic independence can vary. Whether of IBD, allelic similarity, or phenotypic covariance, comparisons may be made to other genomic regions of the same gametes, or to the same genomic regions in other sets of gametes or diploid individuals. In this review, I present IBD as the framework connecting evolutionary and coalescent theory with the analysis of genetic data observed on individuals. I focus on the high variance of the processes that determine IBD, its changes across the genome, and its impact on observable data.

% * TODO: % Where should we mention the work of Georgia on tree sequence IBD?
% There is also a lot of other IBD work \cite{thompson2013identity} in this field!?

% * TODO: Mention eGRM and varGRM \cite{fan2022genealogical} and their modelling work \citep{link2023tree}. Time-varying genetic relatedness also offers insight into how population structure varies as we look further into the past.
%   * I think their work is effectively a large local tree (from Relate) and then they sum/scale over trees and for each
%     tree they work with branch design matrix to get the local G etc.

% * TODO: Check remaining refs from Ines rice manuscript

% TODO: go over what Zhang et al (ARGneddle) how do they get the G?

% A tree sequence represents the relationships between a set of DNA sequences and provides an efficient way of storing genetic variation data.
% It also provides a natural way to describe genetic relatedness via the ancestral relationships encoded in the tree sequence.

% A bit about trees and tree sequences in genetics - an "object/concept" that can bring together disparate areas of genetics -  molecular genetics, population genetics, quantitative genetics, ... I WILL NEED SOME HELP HERE - would like to connect inheritance of genomes/chromosomes between generations, resulting trees (these can skip generations if no data), and pedigrees and then wrap

% TODO: \cite{schraiber2024unifying} drew connections between pedigree,
%       genomic, phylo, and ARG/tree NRMs
%       https://www.biorxiv.org/content/10.1101/2024.02.10.579721v1.full.pdf
%       https://read.readwise.io/read/01hpghf1ejvxtwzw26b14bk0zd

% --- GAP ---------------------------------------------------------------------

\paragraph{GAP: TODO}

% link to portability of genomic analyses across a wide-spectrum of ancestries - see citations in the discussion

Point to a scope for improvement beyong eGRM \citep{fan2022genealogical} by leveraging tree sequence data structure, as well as to have site vs branch options (cite Peter's duality paper CITE and tree sequence stats GENETICS paper \citep{ralph2020efficiently}, centered or non-centered as well as scaled or not

% --- AIM ---------------------------------------------------------------------

\paragraph{The aim of this contribution is an attempt to harmonise different notions of genetic relatedness by working with the tree sequence encoding of ancestral recombination graphs.}

% We demonstrate this using simple examples and finally perform a principal component analysis of X:
%   * a unified genealogy of modern and ancient genomes \cite{wohns2022}
%   * maybe better to use French Canadians (CITE) since we have pedigree there too
%   * alternative would be livestock example, but Brieuc would prefer human example
% All computations are demonstrated using Python tskit library

% covariance of trait processes evolving along the ARG

% \textcolor{red}{TODO: we should build a small example - probably in methods - the example would show a, say, 3-4 generations long pedigree with a manageable number of individuals, including tree sequence and then we can show pedigree, standard genomic and our tree-sequence based versions of relationship matrix}

% --- TERMINOLOGY -------------------------------------------------------------

Wright: relationship coefficient - correlation

Wright/Henderson: numerator relationship coefficient - covariance coefficient

Wright: inbreeding coefficient - correlation \& covariance coefficient

Malecot?: kinship - ???

Falconer: coancestry - ???

???: relatedness - ???

% * TODO: Mention in discussion
%         "Recently, \cite{schraiber2024unifying} have also drew parallels between
%         these definitions to study the estimation of genome-wide associations in
%         presence of population structure and associated environmental confounding."
% 
% * TODO: Cite Me (n eff segments) work, which shows that working with many more segments can actually
%         require more data to estimate them!?
%         Exploring the statistical nature of independent chromosome segments
%         https://doi.org/10.1016/j.livsci.2023.105207
%
% * TODO: Cite in discussion the epistatic drift paper - we should be able to
%         capture a bit of that with mutation effects on ARGs?

\bibliography{references_long}

\end{document}