
$y_i = \mu + g_i + e_i$ Fisher's quantitative genetic model

$y_i$ phenotypic value of individual $i$

$\mu$ population mean

$g_i \sim \N\left(0,\sigma_g^2\right)$

$\sigma_g^2$ genetic variance

$e_i \sim \N\left(0,\sigma_e^2\right)$

$\sigma_e^2$ environmental variance

$R_{i,j} = \Cor\left(g_i, g_j\right)$ Wright's relationship coefficient between individuals $i$ and $j$

$g_i = g_{i,1} + g_{i,2}$ genetic value of a diploid individual under additive gene action

$\Var\left(g_i\right)=\left(1 + F_i\right)\sigma_g^2$

$K_{(i,1),(i,2)} = F_i$ measures ``similarity'' between the two genomes of individual $i$.

$\Cov\left(g_i, g_j\right)=2K_{i,j}\sigma_g^2$

$K_{i,j}$ is similarity between genomes of individual $i$ and genomes of individual $j$, which in diploids gives
$K_{i,j} = \frac{1}{4}\left(
  K_{(i,1), (j,1)} +
  K_{(i,1), (j,2)} +
  K_{(i,2), (j,1)} +
  K_{(i,2), (j,2)}\right)$

$R_{i,j} = \nicefrac{2K_{i,j}}
                    {\sqrt{\left(1 + F_i\right)\left(1 + F_j\right)}}$ with $2K_{i,j}$ referred to as
the \textit{numerator relationship coefficient}

$\Cor\left(g_{i,1}, g_{i,2}\right) = K_{(i,1),(i,2)} = F_i$ Wright's inbreeding coefficient of individual $i$

$F_i=K_{(i,1),(i,2)}=K_{m(i),f(i)}$

$\bsymb{K}$ kinship matrix (probability coefficient!? TODO)

$\bsymb{G} = 2\bsymb{K}$ numerator relationship matrix (NRM; covariance coefficients)

$\bsymb{R}$ Wright's relationship matrix (correlation coefficients)

$g_i = \nicefrac{1}{2}g_{m(i)} + \nicefrac{1}{2}g_{f(i)} + r_i$ Wright's pedigree model

$r_i$ Mendelian sampling deviation

$w_{i,l}$ genotype of individual $i$ at locus $l$

$w_{i,l,k}$ $k$-th allele of individual $i$ at locus $l$ (bi-allelic locus has values $0$ and $1$)

$w_{i,l} = w_{i,l,1} + w_{i,l,2}$ with values $0$, $1$, or $2$

$\bsymb{w}_{i,k}$ multi-locus row vector of alleles (=haplotype) of individual $i$ inherited from parent $k$

$\bsymb{w}_{i}$ multi-locus row vector of genotypes of individual $i$

$\E\left(w_{i,l,k}\right)=p_l$

$q_l = 1 - p_l$

$\E\left(w_{i,l}\right)=2p_l$

$\Var\left(w_{i,l,k}\right) = q_l p_l$

$\Var\left(w_{i,l}\right) = 2 q_l p_l (1 + F_l)$.

$\Cov\left(w_{i,l,k}, w_{j,l,k}\right) = q_l p_l F_l$

identity by state (IBS; where two alleles are identical in their DNA sequence)

identity by descent (IBD; where two alleles are identical due to descending from the same ancestral DNA sequence).

$F_i = \Prob\left(w_{i,l,1} \equiv w_{i,l,2}\right)$ Malecot'consaguinity (=inbreeding) coefficient for individual $i$

$K_{i,j} = \Prob\left(w_{i,l,*} \equiv w_{j,l,*}\right)$ Malecot's parente (=kinship) coefficient between individuals $i$ and $j$

$F_{IT}$ relatedness of individuals (I) with respect to the distant/total/ancestral/etc. reference population (T)

$F_{ST}$ relatedness of sub-populations (S) with respect to the distant/total/top/ancestral/etc. reference population (T)

$F_{IS}$ relatedness of individuals (I) with respect to the current/sub-population/etc. reference population (S)

$F_{IT} = F_{IS} + (1 - F_{IS}) F_{ST}$

$1 - F_{IT} = \left(1 - F_{IS}\right)\left(1 - F_{ST}\right)$

$F_{IT} = F_{IBD} + (1 - F_{IBD}) F_{IBS}$

$1 - F_{IT} = \left(1 - F_{IBD}\right)\left(1 - F_{IBS}\right)$

$F_l = \nicefrac{H_{e,l} - H_{o,l}}{H_{e,l}} = 1 - \nicefrac{H_{o,l}}{H_{e,l}}$ Li & Horvitz (1953) estimator of population inbreeding coefficient

$\bsymb{G}_{x}$ NRM based on a source of information or a method $x$

$\bsymb{G}^{ped}$ pedigree-based NRM

$\bsymb{G}^{m}$ IBS GNRM

$\bsymb{G}^{c}$ genome-based NRM from VanRaden (2008) - the ratio of sums estimator (centering genotypes)

$\bsymb{G}_{s}$ genome-based NRM popularised by Yang et al. (2010)  - the average of ratios (standardizing genotypes)

$\bsymb{G}_{u}$ genome-based NRM from Speed et al. (2012) - the average of weighted ratios (uncorrelating genotypes)
